diff --git a/my-admin/src/App.js b/my-admin/src/App.js
index 7e261ca..4f762c8 100644
--- a/my-admin/src/App.js
+++ b/my-admin/src/App.js
@@ -1,28 +1,4 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React from 'react';
+import { HydraAdmin } from '@api-platform/admin';
 
-class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
-          </p>
-          <a
-            className="App-link"
-            href="https://reactjs.org"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Learn React
-          </a>
-        </header>
-      </div>
-    );
-  }
-}
-
-export default App;
+export default () => <HydraAdmin entrypoint="http://localhost:8000/api"/>; // Replace with your own API entrypoint
