diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 09fa34d..83a39cc 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -4,6 +4,8 @@ namespace App\Entity;
 
 use ApiPlatform\Core\Annotation\ApiResource;
 use ApiPlatform\Core\Annotation\ApiProperty;
+use ApiPlatform\Core\Annotation\ApiFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\BooleanFilter;
 use Symfony\Component\Serializer\Annotation\Groups;
 use Carbon\Carbon;
 use Doctrine\ORM\Mapping as ORM;
@@ -17,6 +19,7 @@ use Doctrine\ORM\Mapping as ORM;
  *     collectionOperations={"get", "post"},
  *     itemOperations={"get", "put"}
  * )
+ * @ApiFilter(BooleanFilter::class, properties={"isPublished"})
  */
 class CheeseListing
 {
