diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 9e1979f..064eca3 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -27,7 +27,10 @@ use Symfony\Component\Validator\Constraints as Assert;
  *     },
  *     collectionOperations={
  *          "get",
- *          "post"={"access_control"="is_granted('ROLE_USER')"}
+ *          "post"={
+ *              "access_control"="is_granted('ROLE_USER')",
+ *              "denormalization_context"={"groups"={"cheese_listing:write", "cheese_listing:post:collection"}},
+ *          }
  *     },
  *     shortName="cheeses",
  *     normalizationContext={"groups"={"cheese_listing:read"}, "swagger_definition_name"="Read"},
@@ -97,7 +100,7 @@ class CheeseListing
     /**
      * @ORM\ManyToOne(targetEntity="App\Entity\User", inversedBy="cheeseListings")
      * @ORM\JoinColumn(nullable=false)
-     * @Groups({"cheese_listing:read", "cheese_listing:write"})
+     * @Groups({"cheese_listing:read", "cheese_listing:post:collection"})
      */
     private $owner;
 
