diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 0f4fa89..b3a850d 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -95,8 +95,7 @@ class CheeseListing
     /**
      * @ORM\ManyToOne(targetEntity="App\Entity\User", inversedBy="cheeseListings")
      * @ORM\JoinColumn(nullable=false)
-     * @Groups({"cheese:read", "cheese:write"})
-     * @Assert\Valid()
+     * @Groups({"cheese:read", "cheese:collection:post"})
      */
     private $owner;
 
diff --git a/src/Entity/User.php b/src/Entity/User.php
index 9ce01a1..59ced90 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -66,7 +66,7 @@ class User implements UserInterface
 
     /**
      * @ORM\Column(type="string", length=255, unique=true)
-     * @Groups({"user:read", "user:write", "cheese:item:get", "cheese:write"})
+     * @Groups({"user:read", "user:write", "cheese:item:get"})
      * @Assert\NotBlank()
      */
     private $username;
