diff --git a/webpack.config.js b/webpack.config.js
index d85c6c4..7563419 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -61,8 +61,8 @@ Encore
     //.autoProvidejQuery()
 
     // uncomment if you use API Platform Admin (composer req api-admin)
-    //.enableReactPreset()
-    //.addEntry('admin', './assets/js/admin.js')
+    .enableReactPreset()
+    .addEntry('admin', './assets/js/admin.js')
 ;
 
 module.exports = Encore.getWebpackConfig();
