diff --git a/src/Doctrine/CheeseListingSetOwnerListener.php b/src/Doctrine/CheeseListingSetOwnerListener.php
index 16483a5..51527a9 100644
--- a/src/Doctrine/CheeseListingSetOwnerListener.php
+++ b/src/Doctrine/CheeseListingSetOwnerListener.php
@@ -3,10 +3,25 @@
 namespace App\Doctrine;
 
 use App\Entity\CheeseListing;
+use Symfony\Component\Security\Core\Security;
 
 class CheeseListingSetOwnerListener
 {
+    private $security;
+
+    public function __construct(Security $security)
+    {
+        $this->security = $security;
+    }
+
     public function prePersist(CheeseListing $cheeseListing)
     {
+        if ($cheeseListing->getOwner()) {
+            return;
+        }
+
+        if ($this->security->getUser()) {
+            $cheeseListing->setOwner($this->security->getUser());
+        }
     }
 }
