diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 0ef1fcd..0f4fa89 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -28,7 +28,7 @@ use Symfony\Component\Validator\Constraints as Assert;
  *          "get",
  *          "post"={"access_control"="is_granted('ROLE_USER')"}
  *     },
- *     shortName="cheeses",
+ *     shortName="cheese",
  *     attributes={
  *          "pagination_items_per_page"=10,
  *          "formats"={"jsonld", "json", "html", "jsonhal", "csv"={"text/csv"}}
@@ -56,7 +56,7 @@ class CheeseListing
 
     /**
      * @ORM\Column(type="string", length=255)
-     * @Groups({"cheese_listing:read", "cheese_listing:write", "user:read", "user:write"})
+     * @Groups({"cheese:read", "cheese:write", "user:read", "user:write"})
      * @Assert\NotBlank()
      * @Assert\Length(
      *     min=2,
@@ -68,7 +68,7 @@ class CheeseListing
 
     /**
      * @ORM\Column(type="text")
-     * @Groups({"cheese_listing:read"})
+     * @Groups({"cheese:read"})
      * @Assert\NotBlank()
      */
     private $description;
@@ -77,7 +77,7 @@ class CheeseListing
      * The price of this delicious cheese, in cents
      *
      * @ORM\Column(type="integer")
-     * @Groups({"cheese_listing:read", "cheese_listing:write", "user:read", "user:write"})
+     * @Groups({"cheese:read", "cheese:write", "user:read", "user:write"})
      * @Assert\NotBlank()
      */
     private $price;
@@ -95,7 +95,7 @@ class CheeseListing
     /**
      * @ORM\ManyToOne(targetEntity="App\Entity\User", inversedBy="cheeseListings")
      * @ORM\JoinColumn(nullable=false)
-     * @Groups({"cheese_listing:read", "cheese_listing:write"})
+     * @Groups({"cheese:read", "cheese:write"})
      * @Assert\Valid()
      */
     private $owner;
@@ -122,7 +122,7 @@ class CheeseListing
     }
 
     /**
-     * @Groups("cheese_listing:read")
+     * @Groups("cheese:read")
      */
     public function getShortDescription(): ?string
     {
@@ -143,7 +143,7 @@ class CheeseListing
     /**
      * The description of the cheese as raw text.
      *
-     * @Groups({"cheese_listing:write", "user:write"})
+     * @Groups({"cheese:write", "user:write"})
      * @SerializedName("description")
      */
     public function setTextDescription(string $description): self
@@ -173,7 +173,7 @@ class CheeseListing
     /**
      * How long ago in text that this cheese listing was added.
      *
-     * @Groups("cheese_listing:read")
+     * @Groups("cheese:read")
      */
     public function getCreatedAtAgo(): string
     {
diff --git a/src/Entity/User.php b/src/Entity/User.php
index 18cb6bd..0627886 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -66,7 +66,7 @@ class User implements UserInterface
 
     /**
      * @ORM\Column(type="string", length=255, unique=true)
-     * @Groups({"user:read", "user:write", "cheese_listing:item:get", "cheese_listing:write"})
+     * @Groups({"user:read", "user:write", "cheese:item:get", "cheese:write"})
      * @Assert\NotBlank()
      */
     private $username;
