diff --git a/src/Entity/User.php b/src/Entity/User.php
index 2ba0d14..a4eb083 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -27,7 +27,9 @@ use Symfony\Component\Validator\Constraints as Assert;
  *          "get",
  *          "put"={"access_control"="is_granted('ROLE_USER') and object == user"},
  *          "delete"={"access_control"="is_granted('ROLE_ADMIN')"}
- *     }
+ *     },
+ *     normalizationContext={"groups"={"user:read"}},
+ *     denormalizationContext={"groups"={"user:write"}},
  * )
  * @ApiFilter(PropertyFilter::class)
  * @UniqueEntity(fields={"username"})
