diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 5ac0f50..1520d7d 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -38,6 +38,11 @@ class CheeseListing
      */
     private $isStinky;
 
+    public function __construct()
+    {
+        $this->createdAt = new \DateTime();
+    }
+
     public function getId(): ?int
     {
         return $this->id;
@@ -72,13 +77,6 @@ class CheeseListing
         return $this->createdAt;
     }
 
-    public function setCreated(\DateTimeInterface $createdAt): self
-    {
-        $this->createdAt = $createdAt;
-
-        return $this;
-    }
-
     public function getIsStinky(): ?bool
     {
         return $this->isStinky;
