diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 0023e87..afb4bab 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -2,10 +2,12 @@
 
 namespace App\Entity;
 
+use ApiPlatform\Core\Annotation\ApiResource;
 use Doctrine\ORM\Mapping as ORM;
 
 /**
  * @ORM\Entity(repositoryClass="App\Repository\CheeseListingRepository")
+ * @ApiResource()
  */
 class CheeseListing
 {
