diff --git a/src/ApiResource/UserApiResource.php b/src/ApiResource/UserApiResource.php
index 65b5f6e5..b880f9e6 100644
--- a/src/ApiResource/UserApiResource.php
+++ b/src/ApiResource/UserApiResource.php
@@ -5,6 +5,7 @@ namespace App\ApiResource;
 use ApiPlatform\Core\Annotation\ApiFilter;
 use ApiPlatform\Core\Annotation\ApiResource;
 use ApiPlatform\Core\Serializer\Filter\PropertyFilter;
+use Symfony\Component\Serializer\Annotation\Groups;
 
 /**
  * @ApiResource(
@@ -29,5 +30,8 @@ use ApiPlatform\Core\Serializer\Filter\PropertyFilter;
  */
 class UserApiResource
 {
+    /**
+     * @Groups({"user:read", "user:write"})
+     */
     public $email;
 }
