diff --git a/src/ApiResource/UserApiResource.php b/src/ApiResource/UserApiResource.php
index b61200e0..61fb4508 100644
--- a/src/ApiResource/UserApiResource.php
+++ b/src/ApiResource/UserApiResource.php
@@ -44,4 +44,43 @@ class UserApiResource
      * @Groups({"user:read", "user:write"})
      */
     public $email;
+
+    /**
+    * @ORM\Column(type="json")
+    * @Groups({"admin:read", "admin:write"})
+    */
+   private $roles = [];
+
+   /**
+    * @var string The hashed password
+    * @ORM\Column(type="string")
+    */
+   private $password;
+
+   /**
+    * @ORM\Column(type="string", length=255, unique=true)
+    * @Groups({"user:read", "user:write", "cheese:item:get"})
+    * @Assert\NotBlank()
+    */
+   private $username;
+
+   /**
+    * @ORM\OneToMany(targetEntity="App\Entity\CheeseListing", mappedBy="owner", cascade={"persist"}, orphanRemoval=true)
+    * @Groups({"user:write"})
+    * @Assert\Valid()
+    */
+   private $cheeseListings;
+
+   /**
+    * @Groups("user:write")
+    * @SerializedName("password")
+    * @Assert\NotBlank(groups={"create"})
+    */
+   private $plainPassword;
+
+   /**
+    * @ORM\Column(type="string", length=50, nullable=true)
+    * @Groups({"admin:read", "owner:read", "user:write"})
+    */
+   private $phoneNumber;
 }
