diff --git a/src/DataProvider/UserApiResourceDataProvider.php b/src/DataProvider/UserApiResourceDataProvider.php
index 2b34ac77..e2ed0381 100644
--- a/src/DataProvider/UserApiResourceDataProvider.php
+++ b/src/DataProvider/UserApiResourceDataProvider.php
@@ -21,7 +21,7 @@ class UserApiResourceDataProvider implements ContextAwareCollectionDataProviderI
 
     public function getCollection(string $resourceClass, string $operationName = null, array $context = [])
     {
-        $queryBuilder = $repository->createQueryBuilder('o');
+        $queryBuilder = $this->userRepository->createQueryBuilder('o');
         $queryNameGenerator = new QueryNameGenerator();
         foreach ($this->collectionExtensions as $extension) {
             $extension->applyToCollection($queryBuilder, $queryNameGenerator, $resourceClass, $operationName, $context);
@@ -31,7 +31,8 @@ class UserApiResourceDataProvider implements ContextAwareCollectionDataProviderI
             }
         }
 
-        return $queryBuilder->getQuery()->getResult();
+        $users = $queryBuilder->getQuery()->getResult();
+        dd($users);
 
         return array_map(function(User $user) {
             $userResource = new UserApiResource();
