diff --git a/src/ApiResource/UserApiResource.php b/src/ApiResource/UserApiResource.php
index 78e7ffe9..5a18789a 100644
--- a/src/ApiResource/UserApiResource.php
+++ b/src/ApiResource/UserApiResource.php
@@ -50,11 +50,6 @@ class UserApiResource
      */
     public $roles = [];
 
-    /**
-     * @var string The hashed password
-     */
-    public $password;
-
     /**
      * @Groups({"user:read", "user:write", "cheese:item:get"})
      * @Assert\NotBlank()
@@ -69,10 +64,9 @@ class UserApiResource
 
     /**
      * @Groups("user:write")
-     * @SerializedName("password")
      * @Assert\NotBlank(groups={"create"})
      */
-    public $plainPassword;
+    public $password;
 
     /**
      * @Groups({"admin:read", "owner:read", "user:write"})
