diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index 0e4ea1a3d..e30422631 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -10,9 +10,11 @@ class DailyStatsProvider implements CollectionDataProviderInterface, RestrictedD
 {
     public function getCollection(string $resourceClass, string $operationName = null)
     {
-        $stats = new DailyStats();
-        $stats->date = new \DateTime();
-        $stats->totalVisitors = 1000;
+        $stats = new DailyStats(
+            new \DateTime(),
+            1000,
+            []
+        );
 
         return [$stats];
     }
diff --git a/src/Entity/DailyStats.php b/src/Entity/DailyStats.php
index 2e0c1586a..3e52f500d 100644
--- a/src/Entity/DailyStats.php
+++ b/src/Entity/DailyStats.php
@@ -38,6 +38,16 @@ class DailyStats
      */
     public $mostPopularListings;
 
+    /**
+     * @param array|CheeseListing[] $mostPopularListings
+     */
+    public function __construct(\DateTimeInterface $date, int $totalVisitors, array $mostPopularListings)
+    {
+        $this->date = $date;
+        $this->totalVisitors = $totalVisitors;
+        $this->mostPopularListings = $mostPopularListings;
+    }
+
     /**
      * @ApiProperty(identifier=true)
      */
