diff --git a/src/Entity/DailyStats.php b/src/Entity/DailyStats.php
index e94dc8e1b..35863e48a 100644
--- a/src/Entity/DailyStats.php
+++ b/src/Entity/DailyStats.php
@@ -2,6 +2,7 @@
 
 namespace App\Entity;
 
+use ApiPlatform\Core\Annotation\ApiProperty;
 use ApiPlatform\Core\Annotation\ApiResource;
 use ApiPlatform\Core\Action\NotFoundAction;
 
@@ -20,6 +21,9 @@ use ApiPlatform\Core\Action\NotFoundAction;
  */
 class DailyStats
 {
+    /**
+     * @ApiProperty(identifier=true)
+     */
     public $date;
 
     public $totalVisitors;
