diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index a2e35ca20..5c452c918 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -26,7 +26,7 @@ class DailyStatsProvider implements CollectionDataProviderInterface, ItemDataPro
 
     public function getItem(string $resourceClass, $id, string $operationName = null, array $context = [])
     {
-        dd($id);
+        return $this->statsHelper->fetchOne($id);
     }
 
     public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
