diff --git a/src/ApiResource/UserApiResource.php b/src/ApiResource/UserApiResource.php
index 12650e1e..78e7ffe9 100644
--- a/src/ApiResource/UserApiResource.php
+++ b/src/ApiResource/UserApiResource.php
@@ -46,36 +46,36 @@ class UserApiResource
     public $email;
 
     /**
-    * @Groups({"admin:read", "admin:write"})
-    */
-   private $roles = [];
+     * @Groups({"admin:read", "admin:write"})
+     */
+    public $roles = [];
 
-   /**
-    * @var string The hashed password
-    */
-   private $password;
+    /**
+     * @var string The hashed password
+     */
+    public $password;
 
-   /**
-    * @Groups({"user:read", "user:write", "cheese:item:get"})
-    * @Assert\NotBlank()
-    */
-   private $username;
+    /**
+     * @Groups({"user:read", "user:write", "cheese:item:get"})
+     * @Assert\NotBlank()
+     */
+    public $username;
 
-   /**
-    * @Groups({"user:write"})
-    * @Assert\Valid()
-    */
-   private $cheeseListings;
+    /**
+     * @Groups({"user:write"})
+     * @Assert\Valid()
+     */
+    public $cheeseListings;
 
-   /**
-    * @Groups("user:write")
-    * @SerializedName("password")
-    * @Assert\NotBlank(groups={"create"})
-    */
-   private $plainPassword;
+    /**
+     * @Groups("user:write")
+     * @SerializedName("password")
+     * @Assert\NotBlank(groups={"create"})
+     */
+    public $plainPassword;
 
-   /**
-    * @Groups({"admin:read", "owner:read", "user:write"})
-    */
-   private $phoneNumber;
+    /**
+     * @Groups({"admin:read", "owner:read", "user:write"})
+     */
+    public $phoneNumber;
 }
