diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index 442cbba..0e4ea1a 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -4,14 +4,21 @@ namespace App\DataProvider;
 
 use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
 use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use App\Entity\DailyStats;
 
 class DailyStatsProvider implements CollectionDataProviderInterface, RestrictedDataProviderInterface
 {
     public function getCollection(string $resourceClass, string $operationName = null)
     {
+        $stats = new DailyStats();
+        $stats->date = new \DateTime();
+        $stats->totalVisitors = 1000;
+
+        return [$stats];
     }
 
     public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
     {
+        return $resourceClass === DailyStats::class;
     }
 }
