diff --git a/src/ApiResource/UserApiResource.php b/src/ApiResource/UserApiResource.php
index f06c2ad7..694ad732 100644
--- a/src/ApiResource/UserApiResource.php
+++ b/src/ApiResource/UserApiResource.php
@@ -2,10 +2,29 @@
 
 namespace App\ApiResource;
 
+use ApiPlatform\Core\Annotation\ApiFilter;
 use ApiPlatform\Core\Annotation\ApiResource;
+use ApiPlatform\Core\Serializer\Filter\PropertyFilter;
 
 /**
- * @ApiResource()
+ * @ApiResource(
+ *     security="is_granted('ROLE_USER')",
+ *     normalizationContext={"groups"={"user:read"}},
+ *     denormalizationContext={"groups"={"user:write"}},
+ *     collectionOperations={
+ *          "get",
+ *          "post"={
+ *              "security"="is_granted('IS_AUTHENTICATED_ANONYMOUSLY')",
+ *              "validation_groups"={"Default", "create"}
+ *          },
+ *     },
+ *     itemOperations={
+ *          "get",
+ *          "put"={"security"="is_granted('ROLE_USER') and object == user"},
+ *          "delete"={"security"="is_granted('ROLE_ADMIN')"}
+ *     }
+ * )
+ * @ApiFilter(PropertyFilter::class)
  */
 class UserApiResource
 {
diff --git a/src/Entity/User.php b/src/Entity/User.php
index 52ef828d..b8926494 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -2,9 +2,6 @@
 
 namespace App\Entity;
 
-use ApiPlatform\Core\Annotation\ApiFilter;
-use ApiPlatform\Core\Annotation\ApiResource;
-use ApiPlatform\Core\Serializer\Filter\PropertyFilter;
 use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\Collections\Collection;
 use Doctrine\ORM\Mapping as ORM;
@@ -15,24 +12,6 @@ use Symfony\Component\Serializer\Annotation\SerializedName;
 use Symfony\Component\Validator\Constraints as Assert;
 
 /**
- * @ApiResource(
- *     security="is_granted('ROLE_USER')",
- *     normalizationContext={"groups"={"user:read"}},
- *     denormalizationContext={"groups"={"user:write"}},
- *     collectionOperations={
- *          "get",
- *          "post"={
- *              "security"="is_granted('IS_AUTHENTICATED_ANONYMOUSLY')",
- *              "validation_groups"={"Default", "create"}
- *          },
- *     },
- *     itemOperations={
- *          "get",
- *          "put"={"security"="is_granted('ROLE_USER') and object == user"},
- *          "delete"={"security"="is_granted('ROLE_ADMIN')"}
- *     }
- * )
- * @ApiFilter(PropertyFilter::class)
  * @UniqueEntity(fields={"username"})
  * @UniqueEntity(fields={"email"})
  * @ORM\Entity(repositoryClass="App\Repository\UserRepository")
