diff --git a/src/ApiResource/UserApiResource.php b/src/ApiResource/UserApiResource.php
index 5a18789a..6da65d59 100644
--- a/src/ApiResource/UserApiResource.php
+++ b/src/ApiResource/UserApiResource.php
@@ -7,6 +7,7 @@ use ApiPlatform\Core\Annotation\ApiProperty;
 use ApiPlatform\Core\Annotation\ApiResource;
 use ApiPlatform\Core\Serializer\Filter\PropertyFilter;
 use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Serializer\Annotation\SerializedName;
 use Symfony\Component\Validator\Constraints as Assert;
 
 /**
@@ -62,6 +63,12 @@ class UserApiResource
      */
     public $cheeseListings;
 
+    /**
+     * @Groups({"user:read"})
+     * @SerializedName("cheeseListings")
+     */
+    public $publishedCheeseListings;
+
     /**
      * @Groups("user:write")
      * @Assert\NotBlank(groups={"create"})
