diff --git a/src/ApiPlatform/DailyStatsDateFilter.php b/src/ApiPlatform/DailyStatsDateFilter.php
index 430239e14..c989b9530 100644
--- a/src/ApiPlatform/DailyStatsDateFilter.php
+++ b/src/ApiPlatform/DailyStatsDateFilter.php
@@ -9,6 +9,7 @@ class DailyStatsDateFilter implements FilterInterface
 {
     public function apply(Request $request, bool $normalization, array $attributes, array &$context)
     {
+        dd($request->query->all());
     }
 
     public function getDescription(string $resourceClass): array
