diff --git a/src/ApiPlatform/DailyStatsDateFilter.php b/src/ApiPlatform/DailyStatsDateFilter.php
index e79a08f..430239e 100644
--- a/src/ApiPlatform/DailyStatsDateFilter.php
+++ b/src/ApiPlatform/DailyStatsDateFilter.php
@@ -13,5 +13,15 @@ class DailyStatsDateFilter implements FilterInterface
 
     public function getDescription(string $resourceClass): array
     {
+        return [
+            'from' => [
+                'property' => null,
+                'type' => 'string',
+                'required' => false,
+                'openapi' => [
+                    'description' => 'From date e.g. 2020-09-01',
+                ],
+            ]
+        ];
     }
 }
