diff --git a/src/ApiPlatform/DailyStatsDateFilter.php b/src/ApiPlatform/DailyStatsDateFilter.php
index c989b9530..ae386574c 100644
--- a/src/ApiPlatform/DailyStatsDateFilter.php
+++ b/src/ApiPlatform/DailyStatsDateFilter.php
@@ -9,7 +9,13 @@ class DailyStatsDateFilter implements FilterInterface
 {
     public function apply(Request $request, bool $normalization, array $attributes, array &$context)
     {
-        dd($request->query->all());
+        $from = $request->query->get('from');
+
+        if (!$from) {
+            return;
+        }
+
+        dd($from);
     }
 
     public function getDescription(string $resourceClass): array
