diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index 1d2b83f0..900814d9 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -26,11 +26,14 @@ class DailyStatsProvider implements CollectionDataProviderInterface, ItemDataPro
     {
         list($page, $offset, $limit) = $this->pagination->getPagination($resourceClass, $operationName);
 
-        return new DailyStatsPaginator(
+        $paginator = new DailyStatsPaginator(
             $this->statsHelper,
             $page,
             $limit
         );
+        $paginator->setFromDate(new \DateTime('2020-08-30'));
+
+        return $paginator;
     }
 
     public function getItem(string $resourceClass, $id, string $operationName = null, array $context = [])
