diff --git a/src/ApiPlatform/DailyStatsDateFilter.php b/src/ApiPlatform/DailyStatsDateFilter.php
index 48ce49b9..e5c165ec 100644
--- a/src/ApiPlatform/DailyStatsDateFilter.php
+++ b/src/ApiPlatform/DailyStatsDateFilter.php
@@ -10,7 +10,7 @@ class DailyStatsDateFilter implements FilterInterface
     public function apply(string $propertyName, $value, DailyStatsPaginator $dailyStatsPaginator)
     {
         if ($propertyName === 'from') {
-            $fromDate = new \DateTimeImmutable($value);
+            $fromDate = \DateTimeImmutable::createFromFormat('Y-m-d', $value);
 
             $dailyStatsPaginator->setFromDate($fromDate);
         }
