diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index bf28da6e..ae005e7e 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -31,7 +31,6 @@ class DailyStatsProvider implements ContextAwareCollectionDataProviderInterface,
 
     public function getCollection(string $resourceClass, string $operationName = null, array $context = [])
     {
-        dd($context);
         list($page, $offset, $limit) = $this->pagination->getPagination($resourceClass, $operationName);
 
         $paginator = new DailyStatsPaginator(
@@ -40,11 +39,15 @@ class DailyStatsProvider implements ContextAwareCollectionDataProviderInterface,
             $limit
         );
 
-        $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
-        $resourceFilters = $resourceMetadata->getCollectionOperationAttribute($operationName, 'filters', [], true);
-        foreach ($resourceFilters as $resourceFilter) {
-            $filterService = $this->filterLocator->get($resourceFilter);
-            dump($filterService);
+        $requestFilters = $context['filters'] ?? [];
+
+        if ($requestFilters) {
+            $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+            $resourceFilters = $resourceMetadata->getCollectionOperationAttribute($operationName, 'filters', [], true);
+            foreach ($resourceFilters as $resourceFilter) {
+                $filterService = $this->filterLocator->get($resourceFilter);
+                dump($filterService);
+            }
         }
 
         $paginator->setFromDate(new \DateTime('2020-08-30'));
