diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index 900814d9..247e0c74 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -3,6 +3,7 @@
 namespace App\DataProvider;
 
 use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\ContextAwareCollectionDataProviderInterface;
 use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
 use ApiPlatform\Core\DataProvider\Pagination;
 use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
@@ -11,7 +12,7 @@ use App\Entity\DailyStats;
 use App\Repository\CheeseListingRepository;
 use App\Service\StatsHelper;
 
-class DailyStatsProvider implements CollectionDataProviderInterface, ItemDataProviderInterface, RestrictedDataProviderInterface
+class DailyStatsProvider implements ContextAwareCollectionDataProviderInterface, ItemDataProviderInterface, RestrictedDataProviderInterface
 {
     private $statsHelper;
     private $pagination;
@@ -22,8 +23,9 @@ class DailyStatsProvider implements CollectionDataProviderInterface, ItemDataPro
         $this->pagination = $pagination;
     }
 
-    public function getCollection(string $resourceClass, string $operationName = null)
+    public function getCollection(string $resourceClass, string $operationName = null, array $context = [])
     {
+        dd($context);
         list($page, $offset, $limit) = $this->pagination->getPagination($resourceClass, $operationName);
 
         $paginator = new DailyStatsPaginator(
