diff --git a/src/Dto/CheeseListingInput.php b/src/Dto/CheeseListingInput.php
index de80fad4..9ae4f661 100644
--- a/src/Dto/CheeseListingInput.php
+++ b/src/Dto/CheeseListingInput.php
@@ -9,11 +9,13 @@ use Symfony\Component\Serializer\Annotation\SerializedName;
 class CheeseListingInput
 {
     /**
+     * @var string
      * @Groups({"cheese:write", "user:write"})
      */
     public $title;
 
     /**
+     * @var int
      * @Groups({"cheese:write", "user:write"})
      */
     public $price;
