diff --git a/src/Dto/CheeseListingInput.php b/src/Dto/CheeseListingInput.php
new file mode 100644
index 00000000..30035aa2
--- /dev/null
+++ b/src/Dto/CheeseListingInput.php
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Dto;
+
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Serializer\Annotation\SerializedName;
+
+class CheeseListingInput
+{
+    /**
+     * @Groups({"cheese:write", "user:write"})
+     */
+    public $title;
+
+    /**
+     * @Groups({"cheese:write", "user:write"})
+     */
+    public $price;
+
+    /**
+     * @Groups({"cheese:collection:post"})
+     */
+    public $owner;
+
+    /**
+     * @Groups({"cheese:write"})
+     */
+    public $isPublished = false;
+
+    public $description;
+
+    /**
+     * The description of the cheese as raw text.
+     *
+     * @Groups({"cheese:write", "user:write"})
+     * @SerializedName("description")
+     */
+    public function setTextDescription(string $description): self
+    {
+        $this->description = nl2br($description);
+
+        return $this;
+    }
+}
diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 98e75a79..3526428c 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -14,7 +14,6 @@ use App\Validator\IsValidOwner;
 use App\Validator\ValidIsPublished;
 use Carbon\Carbon;
 use Doctrine\ORM\Mapping as ORM;
-use Symfony\Component\Serializer\Annotation\Groups;
 use Symfony\Component\Serializer\Annotation\SerializedName;
 use Symfony\Component\Validator\Constraints as Assert;
 
@@ -71,7 +70,6 @@ class CheeseListing
 
     /**
      * @ORM\Column(type="string", length=255)
-     * @Groups({"cheese:write", "user:write"})
      * @Assert\NotBlank()
      * @Assert\Length(
      *     min=2,
@@ -91,7 +89,6 @@ class CheeseListing
      * The price of this delicious cheese, in cents
      *
      * @ORM\Column(type="integer")
-     * @Groups({"cheese:write", "user:write"})
      * @Assert\NotBlank()
      */
     private $price;
@@ -103,14 +100,12 @@ class CheeseListing
 
     /**
      * @ORM\Column(type="boolean")
-     * @Groups({"cheese:write"})
      */
     private $isPublished = false;
 
     /**
      * @ORM\ManyToOne(targetEntity="App\Entity\User", inversedBy="cheeseListings")
      * @ORM\JoinColumn(nullable=false)
-     * @Groups({"cheese:collection:post"})
      * @IsValidOwner()
      */
     private $owner;
@@ -143,19 +138,6 @@ class CheeseListing
         return $this;
     }
 
-    /**
-     * The description of the cheese as raw text.
-     *
-     * @Groups({"cheese:write", "user:write"})
-     * @SerializedName("description")
-     */
-    public function setTextDescription(string $description): self
-    {
-        $this->description = nl2br($description);
-
-        return $this;
-    }
-
     public function getPrice(): ?int
     {
         return $this->price;
