diff --git a/src/DataTransformer/CheeseListingInputDataTransformer.php b/src/DataTransformer/CheeseListingInputDataTransformer.php
index 3015b7a0..79e36cb4 100644
--- a/src/DataTransformer/CheeseListingInputDataTransformer.php
+++ b/src/DataTransformer/CheeseListingInputDataTransformer.php
@@ -15,7 +15,13 @@ class CheeseListingInputDataTransformer implements DataTransformerInterface
     {
         dump($input, $to, $context);
 
-        return new CheeseListing();
+        $cheeseListing = new CheeseListing($input->title);
+        $cheeseListing->setDescription($input->description);
+        $cheeseListing->setPrice($input->price);
+        $cheeseListing->setOwner($input->owner);
+        $cheeseListing->setIsPublished($input->isPublished);
+
+        return $cheeseListing;
     }
 
     public function supportsTransformation($data, string $to, array $context = []): bool
