diff --git a/src/DataTransformer/CheeseListingToInputDataTransformer.php b/src/DataTransformer/CheeseListingToInputDataTransformer.php
index a618c995..03612c94 100644
--- a/src/DataTransformer/CheeseListingToInputDataTransformer.php
+++ b/src/DataTransformer/CheeseListingToInputDataTransformer.php
@@ -22,7 +22,9 @@ class CheeseListingToInputDataTransformer implements DataTransformerInterface
 
         $cheeseListing->setDescription($input->description);
         $cheeseListing->setPrice($input->price);
-        $cheeseListing->setOwner($input->owner);
+        if ($input->owner !== null) {
+            $cheeseListing->setOwner($input->owner);
+        }
 
         return $cheeseListing;
     }
