diff --git a/src/DataTransformer/CheeseListingInputDataTransformer.php b/src/DataTransformer/CheeseListingInputDataTransformer.php
index 2d63e9baa..116b9c83a 100644
--- a/src/DataTransformer/CheeseListingInputDataTransformer.php
+++ b/src/DataTransformer/CheeseListingInputDataTransformer.php
@@ -12,5 +12,8 @@ class CheeseListingInputDataTransformer implements DataTransformerInterface
 
     public function supportsTransformation($data, string $to, array $context = []): bool
     {
+        dump($data, $to, $context);
+
+        return false;
     }
 }
