diff --git a/src/Dto/CheeseListingInput.php b/src/Dto/CheeseListingInput.php
new file mode 100644
index 00000000..fa11c87e
--- /dev/null
+++ b/src/Dto/CheeseListingInput.php
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Dto;
+
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Serializer\Annotation\SerializedName;
+
+class CheeseListingInput
+{
+    /**
+     * @Groups({"cheese:write", "user:write"})
+     */
+    public $title;
+
+    /**
+     * @Groups({"cheese:write", "user:write"})
+     */
+    public $price;
+
+    /**
+     * @Groups({"cheese:collection:post"})
+     */
+    public $owner;
+
+    public $description;
+
+    /**
+     * The description of the cheese as raw text.
+     *
+     * @Groups({"cheese:write", "user:write"})
+     * @SerializedName("description")
+     */
+    public function setTextDescription(string $description): self
+    {
+        $this->description = nl2br($description);
+
+        return $this;
+    }
+}
diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 8b0b526d..7b5e2276 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -12,8 +12,6 @@ use App\Dto\CheeseListingOutput;
 use App\Validator\IsValidOwner;
 use Carbon\Carbon;
 use Doctrine\ORM\Mapping as ORM;
-use Symfony\Component\Serializer\Annotation\Groups;
-use Symfony\Component\Serializer\Annotation\SerializedName;
 use Symfony\Component\Validator\Constraints as Assert;
 
 /**
@@ -67,7 +65,6 @@ class CheeseListing
 
     /**
      * @ORM\Column(type="string", length=255)
-     * @Groups({"cheese:write", "user:write"})
      * @Assert\NotBlank()
      * @Assert\Length(
      *     min=2,
@@ -87,7 +84,6 @@ class CheeseListing
      * The price of this delicious cheese, in cents
      *
      * @ORM\Column(type="integer")
-     * @Groups({"cheese:write", "user:write"})
      * @Assert\NotBlank()
      */
     private $price;
@@ -105,7 +101,6 @@ class CheeseListing
     /**
      * @ORM\ManyToOne(targetEntity="App\Entity\User", inversedBy="cheeseListings")
      * @ORM\JoinColumn(nullable=false)
-     * @Groups({"cheese:collection:post"})
      * @IsValidOwner()
      */
     private $owner;
@@ -138,19 +133,6 @@ class CheeseListing
         return $this;
     }
 
-    /**
-     * The description of the cheese as raw text.
-     *
-     * @Groups({"cheese:write", "user:write"})
-     * @SerializedName("description")
-     */
-    public function setTextDescription(string $description): self
-    {
-        $this->description = nl2br($description);
-
-        return $this;
-    }
-
     public function getPrice(): ?int
     {
         return $this->price;
