diff --git a/src/Dto/CheeseListingOutput.php b/src/Dto/CheeseListingOutput.php
index e3a2a935d..42ddb378f 100644
--- a/src/Dto/CheeseListingOutput.php
+++ b/src/Dto/CheeseListingOutput.php
@@ -11,7 +11,7 @@ class CheeseListingOutput
     /**
      * The title of this listing
      *
-     * @Groups({"cheese:read"})
+     * @Groups({"cheese:read", "user:read"})
      * @var string
      */
     public $title;
@@ -24,7 +24,7 @@ class CheeseListingOutput
 
     /**
      * @var integer
-     * @Groups({"cheese:read"})
+     * @Groups({"cheese:read", "user:read"})
      */
     public $price;
 
