diff --git a/src/Serializer/Normalizer/UserNormalizer.php b/src/Serializer/Normalizer/UserNormalizer.php
index 902fc95..744674e 100644
--- a/src/Serializer/Normalizer/UserNormalizer.php
+++ b/src/Serializer/Normalizer/UserNormalizer.php
@@ -25,7 +25,7 @@ class UserNormalizer implements ContextAwareNormalizerInterface, CacheableSuppor
     /**
      * @param User $object
      */
-    public function normalize($object, $format = null, array $context = array()): array
+    public function normalize($object, $format = null, array $context = array())
     {
         $isOwner = $this->userIsOwner($object);
         if ($isOwner) {
