diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index e56360c8..03e4707c 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -67,7 +67,7 @@ class CheeseListing
 
     /**
      * @ORM\Column(type="string", length=255)
-     * @Groups({"cheese:read", "cheese:write", "user:read", "user:write"})
+     * @Groups({"cheese:write", "user:write"})
      * @Assert\NotBlank()
      * @Assert\Length(
      *     min=2,
@@ -79,7 +79,6 @@ class CheeseListing
 
     /**
      * @ORM\Column(type="text")
-     * @Groups({"cheese:read"})
      * @Assert\NotBlank()
      */
     private $description;
@@ -88,7 +87,7 @@ class CheeseListing
      * The price of this delicious cheese, in cents
      *
      * @ORM\Column(type="integer")
-     * @Groups({"cheese:read", "cheese:write", "user:read", "user:write"})
+     * @Groups({"cheese:write", "user:write"})
      * @Assert\NotBlank()
      */
     private $price;
@@ -106,7 +105,7 @@ class CheeseListing
     /**
      * @ORM\ManyToOne(targetEntity="App\Entity\User", inversedBy="cheeseListings")
      * @ORM\JoinColumn(nullable=false)
-     * @Groups({"cheese:read", "cheese:collection:post"})
+     * @Groups({"cheese:collection:post"})
      * @IsValidOwner()
      */
     private $owner;
@@ -132,18 +131,6 @@ class CheeseListing
         return $this->description;
     }
 
-    /**
-     * @Groups("cheese:read")
-     */
-    public function getShortDescription(): ?string
-    {
-        if (strlen($this->description) < 40) {
-            return $this->description;
-        }
-
-        return substr($this->description, 0, 40).'...';
-    }
-
     public function setDescription(string $description): self
     {
         $this->description = $description;
@@ -181,16 +168,6 @@ class CheeseListing
         return $this->createdAt;
     }
 
-    /**
-     * How long ago in text that this cheese listing was added.
-     *
-     * @Groups("cheese:read")
-     */
-    public function getCreatedAtAgo(): string
-    {
-        return Carbon::instance($this->getCreatedAt())->diffForHumans();
-    }
-
     public function getIsPublished(): ?bool
     {
         return $this->isPublished;
