diff --git a/src/ApiPlatform/DailyStatsDateFilter.php b/src/ApiPlatform/DailyStatsDateFilter.php
index 94e193a8..9e8b7499 100644
--- a/src/ApiPlatform/DailyStatsDateFilter.php
+++ b/src/ApiPlatform/DailyStatsDateFilter.php
@@ -10,6 +10,13 @@ class DailyStatsDateFilter implements FilterInterface
 {
     public const FROM_FILTER_CONTEXT = 'daily_stats_from';
 
+    private $throwOnInvalid;
+
+    public function __construct(bool $throwOnInvalid = false)
+    {
+        $this->throwOnInvalid = $throwOnInvalid;
+    }
+
     public function apply(Request $request, bool $normalization, array $attributes, array &$context)
     {
         $from = $request->query->get('from');
@@ -21,7 +28,7 @@ class DailyStatsDateFilter implements FilterInterface
         $fromDate = \DateTimeImmutable::createFromFormat('Y-m-d', $from);
 
         // you could optionally return a 400 error
-        if (!$fromDate) {
+        if (!$fromDate && $this->throwOnInvalid) {
             throw new BadRequestHttpException('Invalid "from" date format');
         }
 
