diff --git a/src/DataTransformer/CheeseListingOutputDataTransformer.php b/src/DataTransformer/CheeseListingOutputDataTransformer.php
index 0b4b9f3b7..92d4a0167 100644
--- a/src/DataTransformer/CheeseListingOutputDataTransformer.php
+++ b/src/DataTransformer/CheeseListingOutputDataTransformer.php
@@ -8,9 +8,15 @@ use App\Entity\CheeseListing;
 
 class CheeseListingOutputDataTransformer implements DataTransformerInterface
 {
-    public function transform($object, string $to, array $context = [])
+    /**
+     * @param CheeseListing $cheeseListing
+     */
+    public function transform($cheeseListing, string $to, array $context = [])
     {
-        dd($object, $to);
+        $output = new CheeseListingOutput();
+        $output->title = $cheeseListing->getTitle();
+
+        return $output;
     }
 
     public function supportsTransformation($data, string $to, array $context = []): bool
