diff --git a/src/DataTransformer/CheeseListingOutputDataTransformer.php b/src/DataTransformer/CheeseListingOutputDataTransformer.php
index 90d52d62..043de534 100644
--- a/src/DataTransformer/CheeseListingOutputDataTransformer.php
+++ b/src/DataTransformer/CheeseListingOutputDataTransformer.php
@@ -13,14 +13,7 @@ class CheeseListingOutputDataTransformer implements DataTransformerInterface
      */
     public function transform($cheeseListing, string $to, array $context = [])
     {
-        $output = new CheeseListingOutput();
-        $output->title = $cheeseListing->getTitle();
-        $output->description = $cheeseListing->getDescription();
-        $output->price = $cheeseListing->getPrice();
-        $output->owner = $cheeseListing->getOwner();
-        $output->createdAt = $cheeseListing->getCreatedAt();
-
-        return $output;
+        return CheeseListingOutput::createFromEntity($cheeseListing);
     }
 
     public function supportsTransformation($data, string $to, array $context = []): bool
diff --git a/src/Dto/CheeseListingOutput.php b/src/Dto/CheeseListingOutput.php
index 42ddb378..c757c374 100644
--- a/src/Dto/CheeseListingOutput.php
+++ b/src/Dto/CheeseListingOutput.php
@@ -2,6 +2,7 @@
 
 namespace App\Dto;
 
+use App\Entity\CheeseListing;
 use App\Entity\User;
 use Carbon\Carbon;
 use Symfony\Component\Serializer\Annotation\Groups;
@@ -36,6 +37,18 @@ class CheeseListingOutput
      */
     public $owner;
 
+    public static function createFromEntity(CheeseListing $cheeseListing): self
+    {
+        $output = new CheeseListingOutput();
+        $output->title = $cheeseListing->getTitle();
+        $output->description = $cheeseListing->getDescription();
+        $output->price = $cheeseListing->getPrice();
+        $output->owner = $cheeseListing->getOwner();
+        $output->createdAt = $cheeseListing->getCreatedAt();
+
+        return $output;
+    }
+
     /**
      * @Groups("cheese:read")
      */
