diff --git a/src/DataTransformer/CheeseListingOutputDataTransformer.php b/src/DataTransformer/CheeseListingOutputDataTransformer.php
index 92d4a01..80d93f5 100644
--- a/src/DataTransformer/CheeseListingOutputDataTransformer.php
+++ b/src/DataTransformer/CheeseListingOutputDataTransformer.php
@@ -15,6 +15,8 @@ class CheeseListingOutputDataTransformer implements DataTransformerInterface
     {
         $output = new CheeseListingOutput();
         $output->title = $cheeseListing->getTitle();
+        $output->description = $cheeseListing->getDescription();
+        $output->price = $cheeseListing->getPrice();
 
         return $output;
     }
diff --git a/src/Dto/CheeseListingOutput.php b/src/Dto/CheeseListingOutput.php
index 1185668..40e1229 100644
--- a/src/Dto/CheeseListingOutput.php
+++ b/src/Dto/CheeseListingOutput.php
@@ -13,4 +13,16 @@ class CheeseListingOutput
      * @var string
      */
     public $title;
+
+    /**
+     * @var string
+     * @Groups({"cheese:read"})
+     */
+    public $description;
+
+    /**
+     * @var integer
+     * @Groups({"cheese:read"})
+     */
+    public $price;
 }
