diff --git a/tests/Functional/CheeseListingResourceTest.php b/tests/Functional/CheeseListingResourceTest.php
index 5883a9b8..86008a69 100644
--- a/tests/Functional/CheeseListingResourceTest.php
+++ b/tests/Functional/CheeseListingResourceTest.php
@@ -171,7 +171,6 @@ class CheeseListingResourceTest extends CustomApiTestCase
         $this->assertJsonContains(['hydra:member' => [
             0 => [
                 '@id' => '/api/cheeses/' . $cheeseListing2->getId(),
-                '@type' => 'cheese',
                 'title' => 'cheese2',
                 'description' => 'cheese',
                 'price' => 1000,
