diff --git a/src/Dto/CheeseListingOutput.php b/src/Dto/CheeseListingOutput.php
index be0b875..b282b0c 100644
--- a/src/Dto/CheeseListingOutput.php
+++ b/src/Dto/CheeseListingOutput.php
@@ -2,6 +2,7 @@
 
 namespace App\Dto;
 
+use Carbon\Carbon;
 use Symfony\Component\Serializer\Annotation\Groups;
 
 class CheeseListingOutput
@@ -37,4 +38,14 @@ class CheeseListingOutput
 
         return substr($this->description, 0, 40).'...';
     }
+
+    /**
+     * How long ago in text that this cheese listing was added.
+     *
+     * @Groups("cheese:read")
+     */
+    public function getCreatedAtAgo(): string
+    {
+        return Carbon::instance($this->getCreatedAt())->diffForHumans();
+    }
 }
