diff --git a/src/DataTransformer/CheeseListingOutputDataTransformer.php b/src/DataTransformer/CheeseListingOutputDataTransformer.php
index 7ece22708..0b4b9f3b7 100644
--- a/src/DataTransformer/CheeseListingOutputDataTransformer.php
+++ b/src/DataTransformer/CheeseListingOutputDataTransformer.php
@@ -10,11 +10,11 @@ class CheeseListingOutputDataTransformer implements DataTransformerInterface
 {
     public function transform($object, string $to, array $context = [])
     {
-
+        dd($object, $to);
     }
 
     public function supportsTransformation($data, string $to, array $context = []): bool
     {
-        dd($data, $to);
+        return $data instanceof CheeseListing && $to === CheeseListingOutput::class;
     }
 }
