diff --git a/src/Serializer/Normalizer/CheeseListingInputDenormalizer.php b/src/Serializer/Normalizer/CheeseListingInputDenormalizer.php
index 2137b8bd4..61cbda340 100644
--- a/src/Serializer/Normalizer/CheeseListingInputDenormalizer.php
+++ b/src/Serializer/Normalizer/CheeseListingInputDenormalizer.php
@@ -3,6 +3,7 @@
 namespace App\Serializer\Normalizer;
 
 use App\Dto\CheeseListingInput;
+use App\Entity\CheeseListing;
 use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
 use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
 use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
@@ -36,4 +37,28 @@ class CheeseListingInputDenormalizer implements DenormalizerInterface, Cacheable
     {
         return true;
     }
+
+    private function createDto(array $context): CheeseListingInput
+    {
+        $entity = $context['object_to_populate'] ?? null;
+
+        $dto = new CheeseListingInput();
+
+        // not an edit, so just return an empty DTO
+        if (!$entity) {
+            return $dto;
+        }
+
+        if (!$entity instanceof CheeseListing) {
+            throw new \Exception(sprintf('Unexpected resource class "%s"', get_class($entity)));
+        }
+
+        $dto->title = $entity->getTitle();
+        $dto->price = $entity->getPrice();
+        $dto->description = $entity->getDescription();
+        $dto->owner = $entity->getOwner();
+        $dto->isPublished = $entity->getIsPublished();
+
+        return $dto;
+    }
 }
