diff --git a/src/Serializer/Normalizer/CheeseListingInputDenormalizer.php b/src/Serializer/Normalizer/CheeseListingInputDenormalizer.php
new file mode 100644
index 0000000..13b1ced
--- /dev/null
+++ b/src/Serializer/Normalizer/CheeseListingInputDenormalizer.php
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+class CheeseListingInputDenormalizer implements DenormalizerInterface, CacheableSupportsMethodInterface
+{
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+    }
+
+    public function supportsDenormalization($data, string $type, string $format = null)
+    {
+    }
+
+    public function hasCacheableSupportsMethod(): bool
+    {
+    }
+}
