diff --git a/src/Serializer/SerializerContextBuilder.php b/src/Serializer/SerializerContextBuilder.php
index e14f28d..7d4bdeb 100644
--- a/src/Serializer/SerializerContextBuilder.php
+++ b/src/Serializer/SerializerContextBuilder.php
@@ -22,8 +22,10 @@ final class SerializerContextBuilder implements SerializerContextBuilderInterfac
         $context = $this->decorated->createFromRequest($request, $normalization, $extractedAttributes);
         $resourceClass = $context['resource_class'] ?? null;
 
-        if ($resourceClass === Book::class && isset($context['groups']) && $this->authorizationChecker->isGranted('ROLE_ADMIN') && false === $normalization) {
-            $context['groups'][] = 'admin:input';
+        $isAdmin = $this->authorizationChecker->isGranted('ROLE_ADMIN');
+
+        if (isset($context['groups']) && $isAdmin) {
+            $context['groups'][] = $normalization ? 'admin:read' : 'admin:write';
         }
 
         return $context;
