diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 1d8c420..abbb1e0 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -33,13 +33,19 @@ class UserResourceTest extends CustomApiTestCase
 
         $client->request('PUT', '/api/users/'.$user->getId(), [
             'json' => [
-                'username' => 'newusername'
+                'username' => 'newusername',
+                'roles' => ['ROLE_ADMIN'] // will be ignored
             ]
         ]);
         $this->assertResponseIsSuccessful();
         $this->assertJsonContains([
             'username' => 'newusername'
         ]);
+
+        $em = $this->getEntityManager();
+        /** @var User $user */
+        $user = $em->getRepository(User::class)->find($user->getId());
+        $this->assertEquals(['ROLE_USER'], $user->getRoles());
     }
 
     public function testGetUser()
