diff --git a/src/Entity/User.php b/src/Entity/User.php
index e61789d..494eec1 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -87,6 +87,11 @@ class User implements UserInterface
      */
     private $plainPassword;
 
+    /**
+     * @ORM\Column(type="string", length=50, nullable=true)
+     */
+    private $phoneNumber;
+
     public function __construct()
     {
         $this->cheeseListings = new ArrayCollection();
@@ -219,4 +224,16 @@ class User implements UserInterface
 
         return $this;
     }
+
+    public function getPhoneNumber(): ?string
+    {
+        return $this->phoneNumber;
+    }
+
+    public function setPhoneNumber(?string $phoneNumber): self
+    {
+        $this->phoneNumber = $phoneNumber;
+
+        return $this;
+    }
 }
