diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 6709331..1d8c420 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -2,6 +2,7 @@
 
 namespace App\Tests\Functional;
 
+use App\Entity\User;
 use App\Test\CustomApiTestCase;
 use Hautelook\AliceBundle\PhpUnit\ReloadDatabaseTrait;
 
@@ -40,4 +41,33 @@ class UserResourceTest extends CustomApiTestCase
             'username' => 'newusername'
         ]);
     }
+
+    public function testGetUser()
+    {
+        $client = self::createClient();
+        $user = $this->createUserAndLogIn($client, 'cheeseplease@example.com', 'foo');
+
+        $user->setPhoneNumber('555.123.4567');
+        $em = $this->getEntityManager();
+        $em->flush();
+
+        $client->request('GET', '/api/users/'.$user->getId());
+        $this->assertJsonContains([
+            'username' => 'cheeseplease'
+        ]);
+
+        $data = $client->getResponse()->toArray();
+        $this->assertArrayNotHasKey('phoneNumber', $data);
+
+        // refresh the user & elevate
+        $user = $em->getRepository(User::class)->find($user->getId());
+        $user->setRoles(['ROLE_ADMIN']);
+        $em->flush();
+        $this->logIn($client, 'cheeseplease@example.com', 'foo');
+
+        $client->request('GET', '/api/users/'.$user->getId());
+        $this->assertJsonContains([
+            'phoneNumber' => '555.123.4567'
+        ]);
+    }
 }
\ No newline at end of file
