diff --git a/src/Serializer/AdminGroupsContextBuilder.php b/src/Serializer/AdminGroupsContextBuilder.php
index d056561..3ccdf26 100644
--- a/src/Serializer/AdminGroupsContextBuilder.php
+++ b/src/Serializer/AdminGroupsContextBuilder.php
@@ -20,10 +20,11 @@ final class AdminGroupsContextBuilder implements SerializerContextBuilderInterfa
     public function createFromRequest(Request $request, bool $normalization, ?array $extractedAttributes = null): array
     {
         $context = $this->decorated->createFromRequest($request, $normalization, $extractedAttributes);
-        $resourceClass = $context['resource_class'] ?? null;
 
-        if ($resourceClass === Book::class && isset($context['groups']) && $this->authorizationChecker->isGranted('ROLE_ADMIN') && false === $normalization) {
-            $context['groups'][] = 'admin:input';
+        $isAdmin = $this->authorizationChecker->isGranted('ROLE_ADMIN');
+
+        if (isset($context['groups']) && $isAdmin) {
+            $context['groups'][] = $normalization ? 'admin:read' : 'admin:write';
         }
 
         return $context;
