diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index 03c20faa..24df4d66 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -5,21 +5,32 @@ namespace App\DataProvider;
 use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
 use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
 use App\Entity\DailyStats;
+use App\Repository\CheeseListingRepository;
 
 class DailyStatsProvider implements CollectionDataProviderInterface, RestrictedDataProviderInterface
 {
+    private $cheeseListingRepository;
+
+    public function __construct(CheeseListingRepository $cheeseListingRepository)
+    {
+        $this->cheeseListingRepository = $cheeseListingRepository;
+    }
+
     public function getCollection(string $resourceClass, string $operationName = null)
     {
+        $listings = $this->cheeseListingRepository
+            ->findBy([], [], 5);
+
         $stats = new DailyStats(
             new \DateTime(),
             1000,
-            []
+            $listings
         );
 
         $stats2 = new DailyStats(
             new \DateTime('-1 day'),
             2000,
-            []
+            $listings
         );
 
         return [$stats, $stats2];
