diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index e304226..03c20fa 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -16,7 +16,13 @@ class DailyStatsProvider implements CollectionDataProviderInterface, RestrictedD
             []
         );
 
-        return [$stats];
+        $stats2 = new DailyStats(
+            new \DateTime('-1 day'),
+            2000,
+            []
+        );
+
+        return [$stats, $stats2];
     }
 
     public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
