diff --git a/src/Entity/User.php b/src/Entity/User.php
index 33c5f3d1..31c01e29 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -75,7 +75,7 @@ class User implements UserInterface
 
     /**
      * @ORM\OneToMany(targetEntity="App\Entity\CheeseListing", mappedBy="owner", cascade={"persist"}, orphanRemoval=true)
-     * @Groups({"user:write"})
+     * @Groups({"user:write", "user:read"})
      * @Assert\Valid()
      */
     private $cheeseListings;
