diff --git a/src/Entity/DailyStats.php b/src/Entity/DailyStats.php
index d44545704..804e81df6 100644
--- a/src/Entity/DailyStats.php
+++ b/src/Entity/DailyStats.php
@@ -36,7 +36,7 @@ class DailyStats
     /**
      * The 5 most popular cheese listings from this date!
      *
-     * @var array<CheeseListing>
+     * @var array<CheeseListing>|CheeseListing[]
      * @Groups({"daily-stats:read"})
      */
     public $mostPopularListings;
