diff --git a/src/Entity/User.php b/src/Entity/User.php
index d5560e5..106e5ff 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -76,6 +76,11 @@ class User implements UserInterface
      */
     private $cheeseListings;
 
+    /**
+     * @Groups("user:write")
+     */
+    private $plainPassword;
+
     public function __construct()
     {
         $this->cheeseListings = new ArrayCollection();
@@ -196,4 +201,16 @@ class User implements UserInterface
 
         return $this;
     }
+
+    public function getPlainPassword(): ?string
+    {
+        return $this->plainPassword;
+    }
+
+    public function setPlainPassword(string $plainPassword): self
+    {
+        $this->plainPassword = $plainPassword;
+
+        return $this;
+    }
 }
