diff --git a/src/Entity/User.php b/src/Entity/User.php
index fb3eca8..d4f508f 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -159,7 +159,7 @@ class User implements UserInterface
     public function eraseCredentials()
     {
         // If you store any temporary, sensitive data on the user, clear it here
-        // $this->plainPassword = null;
+        $this->plainPassword = null;
     }
 
     public function setUsername(string $username): self
