diff --git a/src/Entity/User.php b/src/Entity/User.php
index 68b4d07..089758f 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -76,6 +76,11 @@ class User implements UserInterface
      */
     private $cheeseListings;
 
+    /**
+     * @Groups("user:write")
+     */
+    private $plainPassword;
+
     public function __construct()
     {
         $this->cheeseListings = new ArrayCollection();
@@ -196,4 +201,14 @@ class User implements UserInterface
 
         return $this;
     }
+
+    public function getPlainPassword()
+    {
+        return $this->plainPassword;
+    }
+
+    public function setPlainPassword($plainPassword)
+    {
+        $this->plainPassword = $plainPassword;
+    }
 }
