diff --git a/src/Entity/User.php b/src/Entity/User.php
index cc0b9ce..6c132de 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -201,13 +201,15 @@ class User implements UserInterface
         return $this;
     }
 
-    public function getPlainPassword()
+    public function getPlainPassword(): ?string
     {
         return $this->plainPassword;
     }
 
-    public function setPlainPassword($plainPassword)
+    public function setPlainPassword(string $plainPassword): self
     {
         $this->plainPassword = $plainPassword;
+
+        return $this;
     }
 }
