diff --git a/src/Entity/User.php b/src/Entity/User.php
index fb341de..8b0cf8e 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -11,6 +11,7 @@ use Doctrine\ORM\Mapping as ORM;
 use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 use Symfony\Component\Security\Core\User\UserInterface;
 use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Serializer\Annotation\SerializedName;
 use Symfony\Component\Validator\Constraints as Assert;
 
 /**
@@ -77,6 +78,7 @@ class User implements UserInterface
 
     /**
      * @Groups("user:write")
+     * @SerializedName("password")
      */
     private $plainPassword;
 
