diff --git a/src/Entity/User.php b/src/Entity/User.php
index 0b5e6df..fb341de 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -160,7 +160,7 @@ class User implements UserInterface
     public function eraseCredentials()
     {
         // If you store any temporary, sensitive data on the user, clear it here
-        // $this->plainPassword = null;
+        $this->plainPassword = null;
     }
 
     public function setUsername(string $username): self
