diff --git a/src/DataPersister/UserDataPersister.php b/src/DataPersister/UserDataPersister.php
index 4e86345..953e68a 100644
--- a/src/DataPersister/UserDataPersister.php
+++ b/src/DataPersister/UserDataPersister.php
@@ -5,14 +5,17 @@ namespace App\DataPersister;
 use ApiPlatform\Core\DataPersister\DataPersisterInterface;
 use App\Entity\User;
 use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
 
 class UserDataPersister implements DataPersisterInterface
 {
     private $entityManager;
+    private $userPasswordEncoder;
 
-    public function __construct(EntityManagerInterface $entityManager)
+    public function __construct(EntityManagerInterface $entityManager, UserPasswordEncoderInterface $userPasswordEncoder)
     {
         $this->entityManager = $entityManager;
+        $this->userPasswordEncoder = $userPasswordEncoder;
     }
 
     public function supports($data): bool
@@ -20,8 +23,18 @@ class UserDataPersister implements DataPersisterInterface
         return $data instanceof User;
     }
 
+    /**
+     * @param User $data
+     */
     public function persist($data)
     {
+        if ($data->getPlainPassword()) {
+            $data->setPassword(
+                $this->userPasswordEncoder->encodePassword($data, $data->getPlainPassword())
+            );
+            $data->eraseCredentials();
+        }
+
         $this->entityManager->persist($data);
         $this->entityManager->flush();
     }
