diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index ca60256..43ed59b 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -5,6 +5,7 @@ namespace App\Entity;
 use ApiPlatform\Core\Annotation\ApiFilter;
 use ApiPlatform\Core\Annotation\ApiResource;
 use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\BooleanFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\DateFilter;
 use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\Collections\Collection;
 use Doctrine\ORM\Mapping as ORM;
@@ -13,6 +14,7 @@ use Doctrine\ORM\Mapping as ORM;
  * @ORM\Entity(repositoryClass="App\Repository\CheeseListingRepository")
  * @ApiResource()
  * @ApiFilter(BooleanFilter::class, properties={"isStinky"})
+ * @ApiFilter(DateFilter::class, properties={"createdAt"})
  */
 class CheeseListing
 {
