diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index a26c809..0ff2800 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -21,7 +21,7 @@ use Symfony\Component\Validator\Constraints as Assert;
  *     denormalizationContext={"groups"="cheese_listing:input"}
  * )
  * @ApiFilter(BooleanFilter::class, properties={"isPublished"})
- * @ApiFilter(SearchFilter::class, properties={"title": "partial", "price": "exact", "description": "partial"})
+ * @ApiFilter(SearchFilter::class, properties={"title": "partial", "price": "exact", "description": "partial", "owner": "exact", "owner.username": "partial"})
  * @ApiFilter(PropertyFilter::class)
  * @ORM\Entity(repositoryClass="App\Repository\CheeseListingRepository")
  */
