diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index a37d19f..d793e2b 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -87,6 +87,15 @@ class CheeseListing
         return $this->description;
     }
 
+    /**
+     * @Groups({"cheese_listing:output"})
+     */
+    public function getShortDescription(): ?string
+    {
+        return substr($this->description, 0,100);
+    }
+
+
     public function setDescription(string $description): self
     {
         $this->description = '<p>'.$description.'</p>';
