diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index fa15f37..0402c73 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -86,6 +86,11 @@ class CheeseListing
         return $this->description;
     }
 
+    /**
+     * The description of the cheese as raw text.
+     *
+     * @Groups("cheese_listing:write")
+     */
     public function setTextDescription(string $description): self
     {
         $this->description = nl2br($description);
