diff --git a/src/EventSubscriber/SetIsMeOnCurrentUserSubscriber.php b/src/EventSubscriber/SetIsMeOnCurrentUserSubscriber.php
new file mode 100644
index 0000000..44dde3c
--- /dev/null
+++ b/src/EventSubscriber/SetIsMeOnCurrentUserSubscriber.php
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+class SetIsMeOnCurrentUserSubscriber implements EventSubscriberInterface
+{
+    public function onRequestEvent(RequestEvent $event)
+    {
+        // ...
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            RequestEvent::class => 'onRequestEvent',
+        ];
+    }
+}
