diff --git a/src/EventSubscriber/SetIsMeOnCurrentUserSubscriber.php b/src/EventSubscriber/SetIsMeOnCurrentUserSubscriber.php
index 44dde3c..33a37f3 100644
--- a/src/EventSubscriber/SetIsMeOnCurrentUserSubscriber.php
+++ b/src/EventSubscriber/SetIsMeOnCurrentUserSubscriber.php
@@ -2,14 +2,33 @@
 
 namespace App\EventSubscriber;
 
+use App\Entity\User;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\Security\Core\Security;
 
 class SetIsMeOnCurrentUserSubscriber implements EventSubscriberInterface
 {
+    private $security;
+
+    public function __construct(Security $security)
+    {
+        $this->security = $security;
+    }
+
     public function onRequestEvent(RequestEvent $event)
     {
-        // ...
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        /** @var User|null $user */
+        $user = $this->security->getUser();
+        if (!$user) {
+            return;
+        }
+
+        $user->setIsMe(true);
     }
 
     public static function getSubscribedEvents()
