diff --git a/src/Entity/User.php b/src/Entity/User.php
index cec41f7..0233abc 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -100,6 +100,13 @@ class User implements UserInterface
      */
     private $isMe = false;
 
+    /**
+     * Returns true if this user is an MVP
+     *
+     * @Groups({"user:read"})
+     */
+    private $isMvp = false;
+
     public function __construct()
     {
         $this->cheeseListings = new ArrayCollection();
@@ -265,4 +272,14 @@ class User implements UserInterface
     {
         $this->isMe = $isMe;
     }
+
+    public function isMvp(): bool
+    {
+        return $this->isMvp;
+    }
+
+    public function setIsMvp(bool $isMvp)
+    {
+        $this->isMvp = $isMvp;
+    }
 }
