diff --git a/src/Doctrine/UserSetIsMvpListener.php b/src/Doctrine/UserSetIsMvpListener.php
new file mode 100644
index 0000000..1b5318a
--- /dev/null
+++ b/src/Doctrine/UserSetIsMvpListener.php
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Doctrine;
+
+use App\Entity\User;
+
+class UserSetIsMvpListener
+{
+    public function postLoad(User $user)
+    {
+        $user->setIsMvp(strpos($user->getUsername(), 'cheese') !== false);
+    }
+}
