diff --git a/assets/js/components/LoginForm.vue b/assets/js/components/LoginForm.vue
index ab14df4..e743455 100644
--- a/assets/js/components/LoginForm.vue
+++ b/assets/js/components/LoginForm.vue
@@ -39,7 +39,6 @@
                 this.isLoading = true;
                 this.error = '';
 
-                /*
                 axios
                     .post('/login', {
                         email: this.email,
@@ -56,7 +55,6 @@
                     }).finally(() => {
                         this.isLoading = false;
                     })
-                */
             },
         },
     }
