diff --git a/src/Controller/SecurityController.php b/src/Controller/SecurityController.php
index 58ed831..fca3d21 100644
--- a/src/Controller/SecurityController.php
+++ b/src/Controller/SecurityController.php
@@ -13,6 +13,12 @@ class SecurityController extends AbstractController
      */
     public function login()
     {
+        if (!$this->isGranted('IS_AUTHENTICATED_FULLY')) {
+            return $this->json([
+                'error' => 'Invalid login request: check that the Content-Type header is "application/json".'
+            ], 400);
+        }
+
         return $this->json([
             'user' => $this->getUser() ? $this->getUser()->getId() : null]
         );
