diff --git a/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php b/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
index 6ef3d43..5ba71e1 100644
--- a/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
+++ b/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
@@ -2,6 +2,7 @@
 
 namespace App\EventSubscriber;
 
+use App\Entity\User;
 use Lexik\Bundle\JWTAuthenticationBundle\Event\AuthenticationSuccessEvent;
 use Lexik\Bundle\JWTAuthenticationBundle\Events;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
@@ -10,7 +11,12 @@ class JwtAuthenticationSuccessSubscriber implements EventSubscriberInterface
 {
     public function onAuthenticationSuccess(AuthenticationSuccessEvent $authenticationSuccessEvent)
     {
-        
+        $data = $authenticationSuccessEvent->getData();
+        /** @var User $user */
+        $user = $authenticationSuccessEvent->getUser();
+        $data['id'] = $user->getId();
+
+        $authenticationSuccessEvent->setData($data);
     }
 
     public static function getSubscribedEvents()
@@ -19,4 +25,4 @@ class JwtAuthenticationSuccessSubscriber implements EventSubscriberInterface
             Events::AUTHENTICATION_SUCCESS => 'onAuthenticationSuccess'
         ];
     }
-}
\ No newline at end of file
+}
