diff --git a/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php b/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
new file mode 100644
index 0000000..0633cb2
--- /dev/null
+++ b/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class JwtAuthenticationSuccessSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        
+    }
+}
\ No newline at end of file
