diff --git a/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php b/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
index 5ba71e1..ec37e77 100644
--- a/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
+++ b/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
@@ -2,6 +2,7 @@
 
 namespace App\EventSubscriber;
 
+use ApiPlatform\Core\Api\IriConverterInterface;
 use App\Entity\User;
 use Lexik\Bundle\JWTAuthenticationBundle\Event\AuthenticationSuccessEvent;
 use Lexik\Bundle\JWTAuthenticationBundle\Events;
@@ -9,12 +10,19 @@ use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
 class JwtAuthenticationSuccessSubscriber implements EventSubscriberInterface
 {
+    private $iriConverter;
+
+    public function __construct(IriConverterInterface $iriConverter)
+    {
+        $this->iriConverter = $iriConverter;
+    }
+
     public function onAuthenticationSuccess(AuthenticationSuccessEvent $authenticationSuccessEvent)
     {
         $data = $authenticationSuccessEvent->getData();
         /** @var User $user */
         $user = $authenticationSuccessEvent->getUser();
-        $data['id'] = $user->getId();
+        $data['user'] = $this->iriConverter->getIriFromItem($user);
 
         $authenticationSuccessEvent->setData($data);
     }
