diff --git a/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php b/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
index 0633cb2..6ef3d43 100644
--- a/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
+++ b/src/EventSubscriber/JwtAuthenticationSuccessSubscriber.php
@@ -2,12 +2,21 @@
 
 namespace App\EventSubscriber;
 
+use Lexik\Bundle\JWTAuthenticationBundle\Event\AuthenticationSuccessEvent;
+use Lexik\Bundle\JWTAuthenticationBundle\Events;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
 class JwtAuthenticationSuccessSubscriber implements EventSubscriberInterface
 {
-    public static function getSubscribedEvents()
+    public function onAuthenticationSuccess(AuthenticationSuccessEvent $authenticationSuccessEvent)
     {
         
     }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            Events::AUTHENTICATION_SUCCESS => 'onAuthenticationSuccess'
+        ];
+    }
 }
\ No newline at end of file
