diff --git a/config/packages/security.yaml b/config/packages/security.yaml
index f8a6e1d..e1a12f3 100644
--- a/config/packages/security.yaml
+++ b/config/packages/security.yaml
@@ -22,6 +22,9 @@ security:
                 username_path: email
                 password_path: password
 
+            logout:
+                path: app_logout
+
             # activate different ways to authenticate
             # https://symfony.com/doc/current/security.html#firewalls-authentication
 
diff --git a/src/Controller/SecurityController.php b/src/Controller/SecurityController.php
index 7dee470..4482661 100644
--- a/src/Controller/SecurityController.php
+++ b/src/Controller/SecurityController.php
@@ -26,4 +26,12 @@ class SecurityController extends AbstractController
             'Location' => $iriConverter->getIriFromItem($this->getUser())
         ]);
     }
+
+    /**
+     * @Route("/logout", name="app_logout")
+     */
+    public function logout()
+    {
+        throw new \Exception('should not be reached');
+    }
 }
