diff --git a/.env.test b/.env.test
index 414430e..07f1a9b 100644
--- a/.env.test
+++ b/.env.test
@@ -2,3 +2,4 @@
 KERNEL_CLASS='App\Kernel'
 APP_SECRET='s$cretf0rt3st'
 SYMFONY_DEPRECATIONS_HELPER=999999
+DATABASE_URL=mysql://symfony:api-platform@127.0.0.1:3306/cheesetestdb
diff --git a/bin/cheesetests.sh b/bin/cheesetests.sh
new file mode 100755
index 0000000..69c5a3a
--- /dev/null
+++ b/bin/cheesetests.sh
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+# To create the database run once:
+# APP_ENV=test php console doctrine:schema:create;
+APP_ENV=test php console doctrine:schema:drop --force;
+APP_ENV=test php console doctrine:schema:update --force;
+APP_ENV=test php console doctrine:fixtures:load -n;
+php phpunit $@
diff --git a/phpunit.xml.dist b/phpunit.xml.dist
index 134c69e..562c673 100644
--- a/phpunit.xml.dist
+++ b/phpunit.xml.dist
@@ -5,12 +5,13 @@
          xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/6.5/phpunit.xsd"
          backupGlobals="false"
          colors="true"
-         bootstrap="config/bootstrap.php"
+         bootstrap="tests/bootstrap.php"
 >
     <php>
         <ini name="error_reporting" value="-1" />
         <env name="APP_ENV" value="test" />
         <env name="SHELL_VERBOSITY" value="-1" />
+        <env name="BOOTSTRAP_RESET_DATABASE" value="1" />
     </php>
 
     <testsuites>
diff --git a/tests/bootstrap.php b/tests/bootstrap.php
new file mode 100644
index 0000000..bec3a83
--- /dev/null
+++ b/tests/bootstrap.php
@@ -0,0 +1,13 @@
+<?php
+
+if (isset($_ENV['BOOTSTRAP_RESET_DATABASE']) && $_ENV['BOOTSTRAP_RESET_DATABASE'] === '1') {
+    passthru( '
+        APP_ENV=test php bin/console doctrine:schema:drop --force;
+        APP_ENV=test php bin/console doctrine:schema:update --force;
+        APP_ENV=test php bin/console doctrine:fixtures:load -n;
+    '
+    );
+    echo 'Recreated the database schema and loaded the DataFixture' . PHP_EOL;
+}
+
+require __DIR__.'/../config/bootstrap.php';
