diff --git a/tests/Functional/CheeseListingResourceTest.php b/tests/Functional/CheeseListingResourceTest.php
index 5883a9b8..afc8af90 100644
--- a/tests/Functional/CheeseListingResourceTest.php
+++ b/tests/Functional/CheeseListingResourceTest.php
@@ -96,6 +96,27 @@ class CheeseListingResourceTest extends CustomApiTestCase
         CheeseNotificationFactory::repository()->assertCount(1);
     }
 
+    public function testPublishCheeseListingMessengerRPC()
+    {
+        $client = self::createClient();
+        $user = UserFactory::new()->create();
+
+        $cheeseListing = CheeseListingFactory::new()
+            ->withLongDescription()
+            ->create(['owner' => $user]);
+
+        $this->logIn($client, $user);
+        $client->request('POST', '/api/cheeses/'.$cheeseListing->getId().'/publish');
+        $this->assertResponseStatusCodeSame(200);
+        $cheeseListing->refresh();
+        $this->assertTrue($cheeseListing->getIsPublished());
+        CheeseNotificationFactory::repository()->assertCount(1, 'There should be one notification about being published');
+
+        // publishing again should not create a second notification
+        $client->request('POST', '/api/cheeses/'.$cheeseListing->getId().'/publish');
+        CheeseNotificationFactory::repository()->assertCount(1);
+    }
+
     public function testPublishCheeseListingValidation()
     {
         $client = self::createClient();
