diff --git a/src/Serializer/Normalizer/UserNormalizer.php b/src/Serializer/Normalizer/UserNormalizer.php
index 78a409b..dc7f075 100644
--- a/src/Serializer/Normalizer/UserNormalizer.php
+++ b/src/Serializer/Normalizer/UserNormalizer.php
@@ -3,10 +3,11 @@
 namespace App\Serializer\Normalizer;
 
 use App\Entity\User;
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
 use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
 
-class UserNormalizer implements NormalizerInterface
+class UserNormalizer implements NormalizerInterface, CacheableSupportsMethodInterface
 {
     private $normalizer;
 
@@ -40,4 +41,9 @@ class UserNormalizer implements NormalizerInterface
     {
         return mt_rand(0, 10) > 5;
     }
+
+    public function hasCacheableSupportsMethod(): bool
+    {
+        return true;
+    }
 }
