diff --git a/src/Serializer/Normalizer/UserNormalizer.php b/src/Serializer/Normalizer/UserNormalizer.php
index 78a409b..da34406 100644
--- a/src/Serializer/Normalizer/UserNormalizer.php
+++ b/src/Serializer/Normalizer/UserNormalizer.php
@@ -3,17 +3,13 @@
 namespace App\Serializer\Normalizer;
 
 use App\Entity\User;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
 use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
-use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
 
-class UserNormalizer implements NormalizerInterface
+class UserNormalizer implements NormalizerInterface, NormalizerAwareInterface
 {
-    private $normalizer;
-
-    public function __construct(ObjectNormalizer $normalizer)
-    {
-        $this->normalizer = $normalizer;
-    }
+    use NormalizerAwareTrait;
 
     /**
      * @param User $object
