diff --git a/src/Serializer/Normalizer/UserNormalizer.php b/src/Serializer/Normalizer/UserNormalizer.php
index 230674a..775779c 100644
--- a/src/Serializer/Normalizer/UserNormalizer.php
+++ b/src/Serializer/Normalizer/UserNormalizer.php
@@ -26,7 +26,8 @@ class UserNormalizer implements ContextAwareNormalizerInterface, NormalizerAware
      */
     public function normalize($object, $format = null, array $context = array()): array
     {
-        if ($this->userIsOwner($object)) {
+        $isOwner = $this->userIsOwner($object);
+        if ($isOwner) {
             $context['groups'][] = 'owner:read';
         }
 
@@ -34,7 +35,7 @@ class UserNormalizer implements ContextAwareNormalizerInterface, NormalizerAware
 
         $data = $this->normalizer->normalize($object, $format, $context);
 
-        // Here: add, edit, or delete some data
+        $data['isMe'] = $isOwner;
 
         return $data;
     }
