diff --git a/src/Serializer/Normalizer/UserNormalizer.php b/src/Serializer/Normalizer/UserNormalizer.php
new file mode 100644
index 0000000..4f3c95f
--- /dev/null
+++ b/src/Serializer/Normalizer/UserNormalizer.php
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+
+class UserNormalizer implements NormalizerInterface, CacheableSupportsMethodInterface
+{
+    private $normalizer;
+
+    public function __construct(ObjectNormalizer $normalizer)
+    {
+        $this->normalizer = $normalizer;
+    }
+
+    public function normalize($object, $format = null, array $context = array()): array
+    {
+        $data = $this->normalizer->normalize($object, $format, $context);
+
+        // Here: add, edit, or delete some data
+
+        return $data;
+    }
+
+    public function supportsNormalization($data, $format = null): bool
+    {
+        return $data instanceof \App\Entity\BlogPost;
+    }
+
+    public function hasCacheableSupportsMethod(): bool
+    {
+        return true;
+    }
+}
