diff --git a/src/Entity/User.php b/src/Entity/User.php
index 0627886..9ce01a1 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -87,7 +87,7 @@ class User implements UserInterface
 
     /**
      * @ORM\Column(type="string", length=50, nullable=true)
-     * @Groups({"admin:read", "user:write"})
+     * @Groups({"admin:read", "owner:read", "user:write"})
      */
     private $phoneNumber;
 
diff --git a/src/Serializer/Normalizer/UserNormalizer.php b/src/Serializer/Normalizer/UserNormalizer.php
index 4f3c95f..384fd52 100644
--- a/src/Serializer/Normalizer/UserNormalizer.php
+++ b/src/Serializer/Normalizer/UserNormalizer.php
@@ -2,6 +2,7 @@
 
 namespace App\Serializer\Normalizer;
 
+use App\Entity\User;
 use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
 use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
@@ -15,8 +16,15 @@ class UserNormalizer implements NormalizerInterface, CacheableSupportsMethodInte
         $this->normalizer = $normalizer;
     }
 
+    /**
+     * @param User $object
+     */
     public function normalize($object, $format = null, array $context = array()): array
     {
+        if ($this->userIsOwner($object)) {
+            $context['groups'][] = 'owner:read';
+        }
+
         $data = $this->normalizer->normalize($object, $format, $context);
 
         // Here: add, edit, or delete some data
@@ -26,7 +34,12 @@ class UserNormalizer implements NormalizerInterface, CacheableSupportsMethodInte
 
     public function supportsNormalization($data, $format = null): bool
     {
-        return $data instanceof \App\Entity\BlogPost;
+        return $data instanceof User;
+    }
+
+    private function userIsOwner(User $user): bool
+    {
+        return mt_rand(0, 10) > 5;
     }
 
     public function hasCacheableSupportsMethod(): bool
