diff --git a/src/Serializer/Normalizer/UserNormalizer.php b/src/Serializer/Normalizer/UserNormalizer.php
index a56f46d..43520a7 100644
--- a/src/Serializer/Normalizer/UserNormalizer.php
+++ b/src/Serializer/Normalizer/UserNormalizer.php
@@ -27,7 +27,8 @@ class UserNormalizer implements ContextAwareNormalizerInterface, CacheableSuppor
      */
     public function normalize($object, $format = null, array $context = array()): array
     {
-        if ($this->userIsOwner($object)) {
+        $isOwner = $this->userIsOwner($object);
+        if ($isOwner) {
             $context['groups'][] = 'owner:read';
         }
 
@@ -35,7 +36,7 @@ class UserNormalizer implements ContextAwareNormalizerInterface, CacheableSuppor
 
         $data = $this->normalizer->normalize($object, $format, $context);
 
-        // Here: add, edit, or delete some data
+        $data['isMe'] = $isOwner;
 
         return $data;
     }
