diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index 94a407e..1d2b83f 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -4,6 +4,7 @@ namespace App\DataProvider;
 
 use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
 use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\Pagination;
 use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
 use ApiPlatform\Core\Exception\ResourceClassNotSupportedException;
 use App\Entity\DailyStats;
@@ -13,18 +14,22 @@ use App\Service\StatsHelper;
 class DailyStatsProvider implements CollectionDataProviderInterface, ItemDataProviderInterface, RestrictedDataProviderInterface
 {
     private $statsHelper;
+    private $pagination;
 
-    public function __construct(StatsHelper $statsHelper)
+    public function __construct(StatsHelper $statsHelper, Pagination $pagination)
     {
         $this->statsHelper = $statsHelper;
+        $this->pagination = $pagination;
     }
 
     public function getCollection(string $resourceClass, string $operationName = null)
     {
+        list($page, $offset, $limit) = $this->pagination->getPagination($resourceClass, $operationName);
+
         return new DailyStatsPaginator(
             $this->statsHelper,
-            1,
-            3
+            $page,
+            $limit
         );
     }
 
