diff --git a/src/DataProvider/DailyStatsProvider.php b/src/DataProvider/DailyStatsProvider.php
index 5c452c9..62b9990 100644
--- a/src/DataProvider/DailyStatsProvider.php
+++ b/src/DataProvider/DailyStatsProvider.php
@@ -21,7 +21,7 @@ class DailyStatsProvider implements CollectionDataProviderInterface, ItemDataPro
 
     public function getCollection(string $resourceClass, string $operationName = null)
     {
-        return $this->statsHelper->fetchMany();
+        return new DailyStatsPaginator();
     }
 
     public function getItem(string $resourceClass, $id, string $operationName = null, array $context = [])
