diff --git a/src/DataProvider/DailyStatsPaginator.php b/src/DataProvider/DailyStatsPaginator.php
index 52f0848..4bedca0 100644
--- a/src/DataProvider/DailyStatsPaginator.php
+++ b/src/DataProvider/DailyStatsPaginator.php
@@ -30,7 +30,7 @@ class DailyStatsPaginator implements PaginatorInterface, \IteratorAggregate
 
     public function count()
     {
-        return 10;
+        return iterator_count($this->getIterator());
     }
 
     public function getIterator()
