diff --git a/src/DataProvider/DailyStatsPaginator.php b/src/DataProvider/DailyStatsPaginator.php
new file mode 100644
index 0000000..c7f7107
--- /dev/null
+++ b/src/DataProvider/DailyStatsPaginator.php
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\DataProvider;
+
+use ApiPlatform\Core\DataProvider\PaginatorInterface;
+
+class DailyStatsPaginator implements PaginatorInterface
+{
+    public function getLastPage(): float
+    {
+    }
+
+    public function getTotalItems(): float
+    {
+    }
+
+    public function getCurrentPage(): float
+    {
+    }
+
+    public function getItemsPerPage(): float
+    {
+    }
+
+    public function count()
+    {
+    }
+}
