diff --git a/src/DataPersister/UserDataPersister.php b/src/DataPersister/UserDataPersister.php
index a940cc64..ae93043a 100644
--- a/src/DataPersister/UserDataPersister.php
+++ b/src/DataPersister/UserDataPersister.php
@@ -2,12 +2,13 @@
 
 namespace App\DataPersister;
 
+use ApiPlatform\Core\DataPersister\ContextAwareDataPersisterInterface;
 use ApiPlatform\Core\DataPersister\DataPersisterInterface;
 use App\Entity\User;
 use Psr\Log\LoggerInterface;
 use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
 
-class UserDataPersister implements DataPersisterInterface
+class UserDataPersister implements ContextAwareDataPersisterInterface
 {
     private $decoratedDataPersister;
     private $userPasswordEncoder;
@@ -20,7 +21,7 @@ class UserDataPersister implements DataPersisterInterface
         $this->logger = $logger;
     }
 
-    public function supports($data): bool
+    public function supports($data, array $context = []): bool
     {
         return $data instanceof User;
     }
@@ -28,7 +29,7 @@ class UserDataPersister implements DataPersisterInterface
     /**
      * @param User $data
      */
-    public function persist($data)
+    public function persist($data, array $context = [])
     {
         if (!$data->getId()) {
             // take any actions needed for a new user
@@ -47,7 +48,7 @@ class UserDataPersister implements DataPersisterInterface
         return $this->decoratedDataPersister->persist($data);
     }
 
-    public function remove($data)
+    public function remove($data, array $context = [])
     {
         $this->decoratedDataPersister->remove($data);
     }
