diff --git a/src/DataPersister/UserDataPersister.php b/src/DataPersister/UserDataPersister.php
index 95e3758b8..a940cc648 100644
--- a/src/DataPersister/UserDataPersister.php
+++ b/src/DataPersister/UserDataPersister.php
@@ -4,17 +4,20 @@ namespace App\DataPersister;
 
 use ApiPlatform\Core\DataPersister\DataPersisterInterface;
 use App\Entity\User;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
 
 class UserDataPersister implements DataPersisterInterface
 {
     private $decoratedDataPersister;
     private $userPasswordEncoder;
+    private $logger;
 
-    public function __construct(DataPersisterInterface $decoratedDataPersister, UserPasswordEncoderInterface $userPasswordEncoder)
+    public function __construct(DataPersisterInterface $decoratedDataPersister, UserPasswordEncoderInterface $userPasswordEncoder, LoggerInterface $logger)
     {
         $this->decoratedDataPersister = $decoratedDataPersister;
         $this->userPasswordEncoder = $userPasswordEncoder;
+        $this->logger = $logger;
     }
 
     public function supports($data): bool
@@ -27,6 +30,13 @@ class UserDataPersister implements DataPersisterInterface
      */
     public function persist($data)
     {
+        if (!$data->getId()) {
+            // take any actions needed for a new user
+            // send registration email
+            // integrate into some CRM or payment system
+            $this->logger->info(sprintf('User %s just registered! Eureka!', $data->getEmail()));
+        }
+
         if ($data->getPlainPassword()) {
             $data->setPassword(
                 $this->userPasswordEncoder->encodePassword($data, $data->getPlainPassword())
