diff --git a/config/packages/doctrine_migrations.yaml b/config/packages/doctrine_migrations.yaml
index 3bf0fbc..61e6612 100644
--- a/config/packages/doctrine_migrations.yaml
+++ b/config/packages/doctrine_migrations.yaml
@@ -1,5 +1,5 @@
 doctrine_migrations:
-    dir_name: '%kernel.project_dir%/src/Migrations'
-    # namespace is arbitrary but should be different from App\Migrations
-    # as migrations classes should NOT be autoloaded
-    namespace: DoctrineMigrations
+    migrations_paths:
+        # namespace is arbitrary but should be different from App\Migrations
+        # as migrations classes should NOT be autoloaded
+        'DoctrineMigrations': '%kernel.project_dir%/migrations'
diff --git a/src/Migrations/.gitignore b/migrations/.gitignore
similarity index 100%
rename from src/Migrations/.gitignore
rename to migrations/.gitignore
diff --git a/src/Migrations/Version20190508193750.php b/migrations/Version20190508193750.php
similarity index 100%
rename from src/Migrations/Version20190508193750.php
rename to migrations/Version20190508193750.php
diff --git a/src/Migrations/Version20190509185722.php b/migrations/Version20190509185722.php
similarity index 100%
rename from src/Migrations/Version20190509185722.php
rename to migrations/Version20190509185722.php
diff --git a/src/Migrations/Version20190509190403.php b/migrations/Version20190509190403.php
similarity index 100%
rename from src/Migrations/Version20190509190403.php
rename to migrations/Version20190509190403.php
diff --git a/src/Migrations/Version20190515191421.php b/migrations/Version20190515191421.php
similarity index 100%
rename from src/Migrations/Version20190515191421.php
rename to migrations/Version20190515191421.php
diff --git a/symfony.lock b/symfony.lock
index 5607c25..c7a1764 100644
--- a/symfony.lock
+++ b/symfony.lock
@@ -59,16 +59,16 @@
         ]
     },
     "doctrine/doctrine-migrations-bundle": {
-        "version": "1.2",
+        "version": "2.2",
         "recipe": {
             "repo": "github.com/symfony/recipes",
             "branch": "master",
-            "version": "1.2",
-            "ref": "c1431086fec31f17fbcfe6d6d7e92059458facc1"
+            "version": "2.2",
+            "ref": "baaa439e3e3179e69e3da84b671f0a3e4a2f56ad"
         },
         "files": [
             "config/packages/doctrine_migrations.yaml",
-            "src/Migrations/.gitignore"
+            "migrations/.gitignore"
         ]
     },
     "doctrine/event-manager": {
