diff --git a/src/Entity/User.php b/src/Entity/User.php
index e19a82c..5ba7a25 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -59,6 +59,7 @@ class User implements UserInterface
     /**
      * @ORM\OneToMany(targetEntity="App\Entity\CheeseListing", mappedBy="owner", cascade={"persist"})
      * @Groups({"user:read", "user:write"})
+     * @Assert\Valid()
      */
     private $cheeseListings;
 
