diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index ab2c44e..df60563 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -34,7 +34,8 @@ use Symfony\Component\Validator\Constraints as Assert;
  * @ApiFilter(SearchFilter::class, properties={
  *     "title": "partial",
  *     "description": "partial",
- *     "owner": "exact"
+ *     "owner": "exact",
+ *     "owner.username": "partial"
  * })
  * @ApiFilter(RangeFilter::class, properties={"price"})
  * @ApiFilter(PropertyFilter::class)
