diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 49f701e..d1f9858 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -35,7 +35,8 @@ use Symfony\Component\Validator\Constraints as Assert;
  * @ApiFilter(SearchFilter::class, properties={
  *     "title": "partial",
  *     "description": "partial",
- *     "owner": "exact"
+ *     "owner": "exact",
+ *     "owner.username": "partial"
  * })
  * @ApiFilter(RangeFilter::class, properties={"price"})
  * @ApiFilter(PropertyFilter::class)
