diff --git a/src/Entity/User.php b/src/Entity/User.php
index b6879ed..cf99d11 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -51,7 +51,7 @@ class User implements UserInterface
 
     /**
      * @ORM\Column(type="string", length=255, unique=true)
-     * @Groups({"user:read", "user:write"})
+     * @Groups({"user:read", "user:write", "cheese_listing:read"})
      * @Assert\NotBlank()
      */
     private $username;
