diff --git a/src/Entity/User.php b/src/Entity/User.php
index c487628..53cf40f 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -14,6 +14,15 @@ use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 
 /**
  * @ApiResource(
+ *     collectionOperations={
+ *         "get"={"access_control"="is_granted('ROLE_USER')"},
+ *         "post"={"access_control"="is_granted('ROLE_ADMIN')"}
+ *     },
+ *     itemOperations={
+ *         "get"={"access_control"="is_granted('ROLE_USER')"},
+ *         "put"={"access_control"="is_granted('ROLE_USER') and object == user"},
+ *         "delete"={"access_control"="is_granted('ROLE_ADMIN')"}
+ *     },
  *     normalizationContext={"groups"="user:output"},
  *     denormalizationContext={"groups"="user:input"}
  * )
@@ -41,6 +50,7 @@ class User implements UserInterface
 
     /**
      * @ORM\Column(type="json")
+     * @Groups({"user:input"})
      */
     private $roles = [];
 
