diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index db696b4..1e8eabe 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -20,7 +20,10 @@ use Symfony\Component\Validator\Constraints as Assert;
  *          "get"={
  *              "normalization_context"={"groups"={"cheese_listing:read", "cheese_listing:get:item"}},
  *          },
- *          "put"={"access_control"="is_granted('ROLE_USER') and object.getOwner() == user"},
+ *          "put"={
+ *              "access_control"="is_granted('ROLE_USER') and object.getOwner() == user",
+ *              "access_control_message"="Only the creator can edit a cheese listing"
+ *          },
  *          "delete"={"access_control"="is_granted('ROLE_ADMIN')"}
  *     },
  *     collectionOperations={
