diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index b22df25..e314ac1 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -20,7 +20,8 @@ use Symfony\Component\Validator\Constraints as Assert;
  *          "get"={
  *              "normalization_context"={"groups"={"cheese_listing:read", "cheese_listing:get:item"}},
  *          },
- *          "put"
+ *          "put"={"access_control"="is_granted('ROLE_USER')"},
+ *          "delete"={"access_control"="is_granted('ROLE_ADMIN')"}
  *     },
  *     collectionOperations={
  *          "get",
