diff --git a/src/Entity/User.php b/src/Entity/User.php
index 4cdf316..d5560e5 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -15,15 +15,16 @@ use Symfony\Component\Validator\Constraints as Assert;
 
 /**
  * @ApiResource(
+ *     accessControl="is_granted('ROLE_USER')",
  *     collectionOperations={
- *          "get"={"access_control"="is_granted('ROLE_USER')"},
- *          "post"
+ *          "get",
+ *          "post"={"access_control"="is_granted('IS_AUTHENTICATED_ANONYMOUSLY')"},
  *     },
  *     itemOperations={
- *          "get"={"access_control"="is_granted('ROLE_USER')"},
+ *          "get",
  *          "put"={"access_control"="is_granted('ROLE_USER') and object == user"},
  *          "delete"={"access_control"="is_granted('ROLE_ADMIN')"}
- *     }
+ *     },
  *     normalizationContext={"groups"={"user:read"}},
  *     denormalizationContext={"groups"={"user:write"}},
  * )
