diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 222ccc7..62677c0 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -21,7 +21,7 @@ use Symfony\Component\Validator\Constraints as Assert;
  *              "normalization_context"={"groups"={"cheese_listing:read", "cheese_listing:item:get"}},
  *          },
  *          "put"={
- *              "access_control"="is_granted('ROLE_USER') and object.getOwner() == user",
+ *              "access_control"="is_granted('ROLE_USER') and previous_object.getOwner() == user",
  *              "access_control_message"="Only the creator can edit a cheese listing"
  *          },
  *          "delete"={"access_control"="is_granted('ROLE_ADMIN')"}
