diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index d5c9fed..3349e85 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -22,6 +22,10 @@ use Symfony\Component\Validator\Constraints as Assert;
  *          },
  *          "put"
  *     },
+ *     collectionOperations={
+ *          "get",
+ *          "post"
+ *     },
  *     shortName="cheeses",
  *     normalizationContext={"groups"={"cheese_listing:read"}, "swagger_definition_name"="Read"},
  *     denormalizationContext={"groups"={"cheese_listing:write"}, "swagger_definition_name"="Write"},
