diff --git a/src/Serializer/CheeseListingContextBuilder.php b/src/Serializer/CheeseListingContextBuilder.php
index d5d79fc..885b244 100644
--- a/src/Serializer/CheeseListingContextBuilder.php
+++ b/src/Serializer/CheeseListingContextBuilder.php
@@ -3,6 +3,7 @@
 namespace App\Serializer;
 
 use ApiPlatform\Core\Serializer\SerializerContextBuilderInterface;
+use App\Entity\CheeseListing;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
 
@@ -20,6 +21,19 @@ final class CheeseListingContextBuilder implements SerializerContextBuilderInter
     public function createFromRequest(Request $request, bool $normalization, ?array $extractedAttributes = null): array
     {
         $context = $this->decorated->createFromRequest($request, $normalization, $extractedAttributes);
+        $resourceClass = $context['resource_class'] ?? null;
+        // Fake that the user has admin priviliges.
+        $isAdmin = true;
+
+        if ($resourceClass === CheeseListing::class && isset($context['groups']) && $isAdmin === true) {
+            if ($normalization) {
+                $context['groups'][] = 'admin:output';
+            }
+            else {
+                $context['groups'][] = 'admin:input';
+            }
+        }
+
         return $context;
     }
 }
