diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 85c98a3..bab8829 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -5,12 +5,15 @@ namespace App\Entity;
 use ApiPlatform\Core\Annotation\ApiResource;
 use Carbon\Carbon;
 use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Serializer\Annotation\Groups;
 
 /**
  * @ApiResource(
  *     collectionOperations={"get", "post"},
  *     itemOperations={"get", "put"},
- *     shortName="cheeses"
+ *     shortName="cheeses",
+ *     normalizationContext={"groups"={"cheese_listing:output"}},
+ *     denormalizationContext={"groups"={"cheese_listing:input"}}
  * )
  * @ORM\Entity(repositoryClass="App\Repository\CheeseListingRepository")
  */
