diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 38c8b7d..4b67661 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -71,7 +71,14 @@ class CheeseListing
 
     public function setDescription(string $description): self
     {
-        $this->description = $description;
+        $this->description = '<p>'.$description.'</p>';
+
+        return $this;
+    }
+
+    public function setRawDescription(string $rawDescription): self
+    {
+        $this->description = $rawDescription;
 
         return $this;
     }
