diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index b04c3de..6dc72f1 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -60,7 +60,7 @@ class CheeseListing
      */
     private $isPublished = false;
 
-    public function __construct(string $title)
+    public function __construct(string $title = null)
     {
         $this->title = $title;
         $this->createdAt = new \DateTimeImmutable();
