diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index f6a8376..ab6c9e7 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -28,6 +28,7 @@ class CheeseListing
 
     /**
      * @ORM\Column(type="string", length=255)
+     * @Groups({"cheese_listing:output", "cheese_listing:input"})
      */
     private $title;
 
@@ -38,6 +39,7 @@ class CheeseListing
 
     /**
      * @ORM\Column(type="integer")
+     * @Groups({"cheese_listing:output", "cheese_listing:input"})
      */
     private $price;
 
@@ -54,6 +56,7 @@ class CheeseListing
     public function __construct()
     {
         $this->createdAt = new \DateTime();
+        $this->isPublished = false;
     }
 
     public function getId(): ?int
