diff --git a/src/DataPersister/CheeseListingDataPersister.php b/src/DataPersister/CheeseListingDataPersister.php
index ae2cde94..3cebaf12 100644
--- a/src/DataPersister/CheeseListingDataPersister.php
+++ b/src/DataPersister/CheeseListingDataPersister.php
@@ -4,14 +4,17 @@ namespace App\DataPersister;
 
 use ApiPlatform\Core\DataPersister\DataPersisterInterface;
 use App\Entity\CheeseListing;
+use Doctrine\ORM\EntityManagerInterface;
 
 class CheeseListingDataPersister implements DataPersisterInterface
 {
     private $decoratedDataPersister;
+    private $entityManager;
 
-    public function __construct(DataPersisterInterface $decoratedDataPersister)
+    public function __construct(DataPersisterInterface $decoratedDataPersister, EntityManagerInterface $entityManager)
     {
         $this->decoratedDataPersister = $decoratedDataPersister;
+        $this->entityManager = $entityManager;
     }
 
     public function supports($data): bool
@@ -24,6 +27,8 @@ class CheeseListingDataPersister implements DataPersisterInterface
      */
     public function persist($data)
     {
+        $originalData = $this->entityManager->getUnitOfWork()->getOriginalEntityData($data);
+        dump($originalData);
         if ($data->getIsPublished()) {
             // hmm, not enough to know that it was JUST published
         }
