diff --git a/src/Validator/ValidIsPublishedValidator.php b/src/Validator/ValidIsPublishedValidator.php
index 2328e6d3..bb89cd69 100644
--- a/src/Validator/ValidIsPublishedValidator.php
+++ b/src/Validator/ValidIsPublishedValidator.php
@@ -35,9 +35,16 @@ class ValidIsPublishedValidator extends ConstraintValidator
             return;
         }
 
-        // TODO: implement the validation here
-        $this->context->buildViolation($constraint->message)
-            ->setParameter('{{ value }}', $value)
-            ->addViolation();
+        if ($value->getIsPublished()) {
+            // we are publishing!
+
+            if (strlen($value->getDescription()) < 100) {
+                $this->context->buildViolation('Cannot publish: description is too short!')
+                    ->atPath('description')
+                    ->addViolation();
+            }
+
+            return;
+        }
     }
 }
