diff --git a/src/Validator/ValidIsPublishedValidator.php b/src/Validator/ValidIsPublishedValidator.php
index f58023549..e2b76042f 100644
--- a/src/Validator/ValidIsPublishedValidator.php
+++ b/src/Validator/ValidIsPublishedValidator.php
@@ -3,11 +3,19 @@
 namespace App\Validator;
 
 use App\Entity\CheeseListing;
+use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\ConstraintValidator;
 
 class ValidIsPublishedValidator extends ConstraintValidator
 {
+    private $entityManager;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
     public function validate($value, Constraint $constraint)
     {
         /* @var $constraint \App\Validator\ValidIsPublished */
@@ -16,7 +24,10 @@ class ValidIsPublishedValidator extends ConstraintValidator
             throw new \LogicException('Only CheeseListing is supported');
         }
 
-        dd($value);
+        $originalData = $this->entityManager
+            ->getUnitOfWork()
+            ->getOriginalEntityData($value);
+        dd($originalData);
 
         if (null === $value || '' === $value) {
             return;
