diff --git a/tests/Functional/CheeseListingResourceTest.php b/tests/Functional/CheeseListingResourceTest.php
index ab24e578..5977a324 100644
--- a/tests/Functional/CheeseListingResourceTest.php
+++ b/tests/Functional/CheeseListingResourceTest.php
@@ -73,9 +73,9 @@ class CheeseListingResourceTest extends CustomApiTestCase
         $client = self::createClient();
         $user = UserFactory::new()->create();
 
-        $cheeseListing = CheeseListingFactory::new()->create([
-            'owner' => $user,
-        ]);
+        $cheeseListing = CheeseListingFactory::new()
+            ->withLongDescription()
+            ->create(['owner' => $user]);
 
         $this->logIn($client, $user);
         $client->request('PUT', '/api/cheeses/'.$cheeseListing->getId(), [
