diff --git a/tests/Functional/CheeseListingResourceTest.php b/tests/Functional/CheeseListingResourceTest.php
index ad32e91..f40e303 100644
--- a/tests/Functional/CheeseListingResourceTest.php
+++ b/tests/Functional/CheeseListingResourceTest.php
@@ -3,6 +3,7 @@
 namespace App\Tests\Functional;
 
 use App\ApiPlatform\Test\ApiTestCase;
+use App\Entity\User;
 
 class CheeseListingResourceTest extends ApiTestCase
 {
@@ -14,5 +15,23 @@ class CheeseListingResourceTest extends ApiTestCase
             'json' => [],
         ]);
         $this->assertResponseStatusCodeSame(401);
+
+        $user = new User();
+        $user->setEmail('cheeseplease@example.com');
+        $user->setUsername('cheeseplease');
+        $user->setPassword('$argon2id$v=19$m=65536,t=6,p=1$AIC3IESQ64NgHfpVQZqviw$1c7M56xyiaQFBjlUBc7T0s53/PzZCjV56lbHnhOUXx8');
+
+        $em = self::$container->get('doctrine')->getManager();
+        $em->persist($user);
+        $em->flush();
+
+        $client->request('POST', '/login', [
+            'headers' => ['Content-Type' => 'application/json'],
+            'json' => [
+                'email' => 'cheeseplease@example.com',
+                'password' => 'foo'
+            ],
+        ]);
+        $this->assertResponseStatusCodeSame(204);
     }
 }
