diff --git a/src/Test/CustomApiTestCase.php b/src/Test/CustomApiTestCase.php
index d116e9a..393727b 100644
--- a/src/Test/CustomApiTestCase.php
+++ b/src/Test/CustomApiTestCase.php
@@ -13,7 +13,10 @@ class CustomApiTestCase extends ApiTestCase
         $user = new User();
         $user->setEmail($email);
         $user->setUsername(substr($email, 0, strpos($email, '@')));
-        $user->setPassword($password);
+
+        $encoded = self::$container->get('security.password_encoder')
+            ->encodePassword($user, $password);
+        $user->setPassword($encoded);
 
         $em = self::$container->get('doctrine')->getManager();
         $em->persist($user);
diff --git a/tests/Functional/CheeseListingResourceTest.php b/tests/Functional/CheeseListingResourceTest.php
index dbd88df..97bd63e 100644
--- a/tests/Functional/CheeseListingResourceTest.php
+++ b/tests/Functional/CheeseListingResourceTest.php
@@ -19,7 +19,7 @@ class CheeseListingResourceTest extends CustomApiTestCase
         ]);
         $this->assertResponseStatusCodeSame(401);
 
-        $this->createUser('cheeseplease@example.com', '$argon2id$v=19$m=65536,t=6,p=1$AIC3IESQ64NgHfpVQZqviw$1c7M56xyiaQFBjlUBc7T0s53/PzZCjV56lbHnhOUXx8');
+        $this->createUser('cheeseplease@example.com', 'foo');;
         $this->logIn($client, 'cheeseplease@example.com', 'foo');
     }
 }
