diff --git a/src/DataProvider/UserDataProvider.php b/src/DataProvider/UserDataProvider.php
index a359c229..3a49c5c9 100644
--- a/src/DataProvider/UserDataProvider.php
+++ b/src/DataProvider/UserDataProvider.php
@@ -4,6 +4,7 @@ namespace App\DataProvider;
 
 use ApiPlatform\Core\DataProvider\ContextAwareCollectionDataProviderInterface;
 use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use App\Entity\User;
 
 class UserDataProvider implements ContextAwareCollectionDataProviderInterface, RestrictedDataProviderInterface
 {
@@ -13,5 +14,6 @@ class UserDataProvider implements ContextAwareCollectionDataProviderInterface, R
 
     public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
     {
+        return $resourceClass === User::class;
     }
 }
