diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 2094ec51..047a1cb5 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -62,6 +62,9 @@ class UserResourceTest extends CustomApiTestCase
 
         $data = $client->getResponse()->toArray();
         $this->assertArrayNotHasKey('phoneNumber', $data);
+        $this->assertJsonContains([
+            'isMe' => false,
+        ]);
 
         // refresh the user & elevate
         $user->refresh();
@@ -71,7 +74,8 @@ class UserResourceTest extends CustomApiTestCase
 
         $client->request('GET', '/api/users/'.$user->getId());
         $this->assertJsonContains([
-            'phoneNumber' => '555.123.4567'
+            'phoneNumber' => '555.123.4567',
+            'isMe' => true,
         ]);
     }
 }
