diff --git a/src/DataProvider/UserDataProvider.php b/src/DataProvider/UserDataProvider.php
index 3f5e1329..8e9180aa 100644
--- a/src/DataProvider/UserDataProvider.php
+++ b/src/DataProvider/UserDataProvider.php
@@ -18,8 +18,14 @@ class UserDataProvider implements ContextAwareCollectionDataProviderInterface, R
 
     public function getCollection(string $resourceClass, string $operationName = null, array $context = [])
     {
+        /** @var User[] $users */
         $users = $this->collectionDataProvider->getCollection($resourceClass, $operationName, $context);
-        dd($users);
+
+        foreach ($users as $user) {
+            $user->setIsMe(true);
+        }
+
+        return $users;
     }
 
     public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
