diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index 9457067..88c79cd 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -41,7 +41,7 @@ class CheeseListing
      *     max = 50
      * )
      * @ORM\Column(type="string", length=255)
-     * @Groups({"cheese_listing:output", "cheese_listing:input"})
+     * @Groups({"cheese_listing:output", "cheese_listing:input", "user:output"})
      */
     private $title;
 
@@ -107,7 +107,7 @@ class CheeseListing
     }
 
     /**
-     * @Groups({"cheese_listing:output"})
+     * @Groups({"cheese_listing:output", "user:output"})
      */
     public function getShortDescription(): ?string
     {
diff --git a/src/Entity/User.php b/src/Entity/User.php
index fb4c749..bf4d613 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -53,7 +53,7 @@ class User implements UserInterface
     /**
      * @Assert\NotBlank
      * @ORM\Column(type="string", length=255)
-     * @Groups({"user:output", "user:input"})
+     * @Groups({"user:output", "user:input", "cheese_listing:output"})
      */
     private $username;
 
