diff --git a/src/DataProvider/UserDataProvider.php b/src/DataProvider/UserDataProvider.php
index dc4e2c0a..1716afc7 100644
--- a/src/DataProvider/UserDataProvider.php
+++ b/src/DataProvider/UserDataProvider.php
@@ -4,11 +4,12 @@ namespace App\DataProvider;
 
 use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
 use ApiPlatform\Core\DataProvider\ContextAwareCollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\DenormalizedIdentifiersAwareItemDataProviderInterface;
 use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
 use App\Entity\User;
 use Symfony\Component\Security\Core\Security;
 
-class UserDataProvider implements ContextAwareCollectionDataProviderInterface, RestrictedDataProviderInterface
+class UserDataProvider implements ContextAwareCollectionDataProviderInterface, DenormalizedIdentifiersAwareItemDataProviderInterface, RestrictedDataProviderInterface
 {
     private $collectionDataProvider;
     private $security;
@@ -32,6 +33,10 @@ class UserDataProvider implements ContextAwareCollectionDataProviderInterface, R
         return $users;
     }
 
+    public function getItem(string $resourceClass, $id, string $operationName = null, array $context = [])
+    {
+    }
+
     public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
     {
         return $resourceClass === User::class;
