diff --git a/src/DataProvider/UserDataProvider.php b/src/DataProvider/UserDataProvider.php
index 8eb0966..8e9180a 100644
--- a/src/DataProvider/UserDataProvider.php
+++ b/src/DataProvider/UserDataProvider.php
@@ -18,8 +18,12 @@ class UserDataProvider implements ContextAwareCollectionDataProviderInterface, R
 
     public function getCollection(string $resourceClass, string $operationName = null, array $context = [])
     {
+        /** @var User[] $users */
         $users = $this->collectionDataProvider->getCollection($resourceClass, $operationName, $context);
-        dd($users);
+
+        foreach ($users as $user) {
+            $user->setIsMe(true);
+        }
 
         return $users;
     }
