diff --git a/src/Entity/User.php b/src/Entity/User.php
index 6365ea0dd..064f56280 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -258,6 +258,10 @@ class User implements UserInterface
 
     public function getIsMe(): bool
     {
+        if ($this->isMe === null) {
+            throw new \LogicException('The isMe field has not been initialized');
+        }
+
         return $this->isMe;
     }
 
