diff --git a/src/Entity/User.php b/src/Entity/User.php
index b8ebb58..d5405b3 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -2,10 +2,12 @@
 
 namespace App\Entity;
 
+use ApiPlatform\Core\Annotation\ApiResource;
 use Doctrine\ORM\Mapping as ORM;
 use Symfony\Component\Security\Core\User\UserInterface;
 
 /**
+ * @ApiResource()
  * @ORM\Entity(repositoryClass="App\Repository\UserRepository")
  */
 class User implements UserInterface
