diff --git a/src/Entity/User.php b/src/Entity/User.php
index b0fba103..7a3dfcb1 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -9,6 +9,7 @@ use ApiPlatform\Core\Serializer\Filter\PropertyFilter;
 use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\Collections\Collection;
 use Doctrine\ORM\Mapping as ORM;
+use Ramsey\Uuid\Uuid;
 use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 use Symfony\Component\Security\Core\User\UserInterface;
 use Symfony\Component\Serializer\Annotation\Groups;
@@ -114,6 +115,7 @@ class User implements UserInterface
     public function __construct()
     {
         $this->cheeseListings = new ArrayCollection();
+        $this->uuid = Uuid::uuid4();
     }
 
     public function getId(): ?int
