diff --git a/src/Entity/User.php b/src/Entity/User.php
index b8206c8f..63318796 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -48,11 +48,13 @@ class User implements UserInterface
      * @ORM\Id()
      * @ORM\GeneratedValue()
      * @ORM\Column(type="integer")
+     * @ApiProperty(identifier=false)
      */
     private $id;
 
     /**
      * @ORM\Column(type="uuid", unique=true)
+     * @ApiProperty(identifier=true)
      */
     private $uuid;
 
