diff --git a/src/DataTransformer/CheeseListingToInputDataTransformer.php b/src/DataTransformer/CheeseListingToInputDataTransformer.php
index cd28c111..75e6dede 100644
--- a/src/DataTransformer/CheeseListingToInputDataTransformer.php
+++ b/src/DataTransformer/CheeseListingToInputDataTransformer.php
@@ -24,7 +24,11 @@ class CheeseListingToInputDataTransformer implements DataTransformerInterface
     {
         $isUpdate = isset($context[AbstractItemNormalizer::OBJECT_TO_POPULATE]);
 
-        $this->validator->validate($input);
+        $groups = [];
+        if (!$isUpdate) {
+            $groups[] = 'new';
+        }
+        $this->validator->validate($input, ['groups' => $groups]);
 
         if ($isUpdate) {
             $cheeseListing = $context[AbstractItemNormalizer::OBJECT_TO_POPULATE];
