diff --git a/src/Dto/CheeseListingInput.php b/src/Dto/CheeseListingInput.php
index 24d466df..d848cd86 100644
--- a/src/Dto/CheeseListingInput.php
+++ b/src/Dto/CheeseListingInput.php
@@ -59,8 +59,8 @@ class CheeseListingInput
             $cheeseListing = new CheeseListing($this->title);
         }
 
-        $cheeseListing->setDescription($this->description);
-        $cheeseListing->setPrice($this->price);
+        $cheeseListing->setDescription((string) $this->description);
+        $cheeseListing->setPrice((int) $this->price);
         $cheeseListing->setOwner($this->owner);
         $cheeseListing->setIsPublished($this->isPublished);
 
