diff --git a/src/Entity/User.php b/src/Entity/User.php
index 5c1177f..2a52020 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -19,7 +19,10 @@ use Symfony\Component\Validator\Constraints as Assert;
  *     accessControl="is_granted('ROLE_USER')",
  *     collectionOperations={
  *          "get",
- *          "post"={"access_control"="is_granted('IS_AUTHENTICATED_ANONYMOUSLY')"},
+ *          "post"={
+ *              "access_control"="is_granted('IS_AUTHENTICATED_ANONYMOUSLY')",
+ *              "validation_groups"={"Default", "create"}
+ *          },
  *     },
  *     itemOperations={
  *          "get",
@@ -79,7 +82,7 @@ class User implements UserInterface
     /**
      * @Groups("user:write")
      * @SerializedName("password")
-     * @Assert\NotBlank()
+     * @Assert\NotBlank(groups={"create"})
      */
     private $plainPassword;
 
