diff --git a/src/Dto/CheeseListingInput.php b/src/Dto/CheeseListingInput.php
index 075f27a8..d7596fd8 100644
--- a/src/Dto/CheeseListingInput.php
+++ b/src/Dto/CheeseListingInput.php
@@ -13,7 +13,7 @@ class CheeseListingInput
     /**
      * @var string
      * @Groups({"cheese:write", "user:write"})
-     * @Assert\NotBlank()
+     * @Assert\NotBlank(groups={"new"})
      * @Assert\Length(
      *     min=2,
      *     max=50,
@@ -25,7 +25,7 @@ class CheeseListingInput
     /**
      * @var int
      * @Groups({"cheese:write", "user:write"})
-     * @Assert\NotBlank()
+     * @Assert\NotBlank(groups={"new"})
      */
     public $price;
 
@@ -37,7 +37,7 @@ class CheeseListingInput
     public $owner;
 
     /**
-     * @Assert\NotBlank()
+     * @Assert\NotBlank(groups={"new"})
      */
     public $description;
 
