diff --git a/src/Entity/CheeseListing.php b/src/Entity/CheeseListing.php
index d172e186..31a7d85a 100644
--- a/src/Entity/CheeseListing.php
+++ b/src/Entity/CheeseListing.php
@@ -12,7 +12,6 @@ use App\Dto\CheeseListingOutput;
 use App\Dto\CheeseListingInput;
 use Carbon\Carbon;
 use Doctrine\ORM\Mapping as ORM;
-use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 
 /**
  * @ApiResource(
@@ -54,7 +53,6 @@ use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
  * @ApiFilter(PropertyFilter::class)
  * @ORM\Entity(repositoryClass="App\Repository\CheeseListingRepository")
  * @ORM\EntityListeners({"App\Doctrine\CheeseListingSetOwnerListener"})
- * @UniqueEntity(fields={"title"})
  */
 class CheeseListing
 {
