diff --git a/src/Dto/CheeseListingInput.php b/src/Dto/CheeseListingInput.php
index c54e857e..075f27a8 100644
--- a/src/Dto/CheeseListingInput.php
+++ b/src/Dto/CheeseListingInput.php
@@ -47,7 +47,7 @@ class CheeseListingInput
      * @Groups({"cheese:write", "user:write"})
      * @SerializedName("description")
      */
-    public function setTextDescription(string $description): self
+    public function setTextDescription($description): self
     {
         $this->description = nl2br($description);
 
