diff --git a/src/Security/Voter/CheeseListingVoter.php b/src/Security/Voter/CheeseListingVoter.php
index c1de011..e3bd440 100644
--- a/src/Security/Voter/CheeseListingVoter.php
+++ b/src/Security/Voter/CheeseListingVoter.php
@@ -5,10 +5,18 @@ namespace App\Security\Voter;
 use App\Entity\CheeseListing;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\Security;
 use Symfony\Component\Security\Core\User\UserInterface;
 
 class CheeseListingVoter extends Voter
 {
+    private $security;
+
+    public function __construct(Security $security)
+    {
+        $this->security = $security;
+    }
+
     protected function supports($attribute, $subject)
     {
         // replace with your own logic
@@ -34,6 +42,10 @@ class CheeseListingVoter extends Voter
                     return true;
                 }
 
+                if ($this->security->isGranted('ROLE_ADMIN')) {
+                    return true;
+                }
+
                 return false;
         }
 
